% Template for PLoS
% Version 2.0 July 2014
%
% To compile to pdf, run:
% latex plos.template
% bibtex plos.template
% latex plos.template
% latex plos.template
% dvipdf plos.template
%
% % % % % % % % % % % % % % % % % % % % % %
%
% -- FIGURES AND TABLES
%
% DO NOT INCLUDE GRAPHICS IN YOUR MANUSCRIPT
% - Figures should be uploaded separately from your manuscript file. 
% - Figures generated using LaTeX should be extracted and removed from the PDF before submission. 
% - Figures containing multiple panels/subfigures must be combined into one image file before submission.
% See http://www.plosone.org/static/figureGuidelines for PLOS figure guidelines.
%
% % % % % % % % % % % % % % % % % % % % % % % %
%
% -- EQUATIONS, MATH SYMBOLS, SUBSCRIPTS, AND SUPERSCRIPTS
%
% IMPORTANT
% Below are a few tips to help format your equations and other special characters according to our specifications. For more tips to help reduce the possibility of formatting errors during conversion, please see our LaTeX guidelines at http://www.plosone.org/static/latexGuidelines
%
% Please be sure to include all portions of an equation in the math environment, and for any superscripts or subscripts also include the base number/text. For example, use $mathrm{mm}^2$ instead of mm$^2$ (do not use \textsuperscript command).
%
% For inline equations, please do not include punctuation within the math environment unless this is part of the equation.
%
% For spaces within the math environment please use the \; or \: commands, even within \text{} (do not use smaller spacing as this does not convert well).
%
%
% % % % % % % % % % % % % % % % % % % % % % % %



\documentclass[10pt]{article}

% amsmath package, useful for mathematical formulas
\usepackage{amsmath}
% amssymb package, useful for mathematical symbols
\usepackage{amssymb}

% cite package, to clean up citations in the main text. Do not remove.
\usepackage{cite}

\usepackage{hyperref}

% line numbers
\usepackage{lineno}

% ligatures disabled
\usepackage{microtype}
\DisableLigatures[f]{encoding = *, family = * }

% rotating package for sideways tables
%\usepackage{rotating}

% If you wish to include algorithms, please use one of the packages below. Also, please see the algorithm section of our LaTeX guidelines (http://www.plosone.org/static/latexGuidelines) for important information about required formatting.
%\usepackage{algorithmic}
%\usepackage{algorithmicx}

% Use doublespacing - comment out for single spacing
%\usepackage{setspace} 
%\doublespacing


% Text layout
\topmargin 0.0cm
\oddsidemargin 0.5cm
\evensidemargin 0.5cm
\textwidth 16cm 
\textheight 21cm

% Bold the 'Figure #' in the caption and separate it with a period
% Captions will be left justified
\usepackage[labelfont=bf,labelsep=period,justification=raggedright]{caption}

% Use the PLoS provided BiBTeX style
\bibliographystyle{plos2009}

% Remove brackets from numbering in List of References
\makeatletter
\renewcommand{\@biblabel}[1]{\quad#1.}
\makeatother


% Leave date blank
\date{}

\pagestyle{myheadings}

%% Include all macros below. Please limit the use of macros.

\newcommand\D{\mathrm{d}}

\newcommand{\Fig}{Fig.}
\newcommand{\Figs}{Fig.} % It is the same for PLOS

\newcommand{\citep}{\cite}
\newcommand{\citet}{\citep}

%% END MACROS SECTION


\begin{document}


% Title must be 150 characters or less
\begin{flushleft}
{\Large
\textbf{Dynamic Epistasis under Varying Environmental Perturbations}
}
% Insert Author names, affiliations and corresponding author email.
\\
Brandon Barker$^{1,\ast}$, 
Lin Xu$^{2,\ast}$, 
Zhenglong Gu$^{3,4,\dag}$
\\
\bf{1} Center for Advanced Computing, Cornell University, Ithaca, New
York, United States of America
\\
\bf{2} Division of Hematology/Oncology, Department of Pediatrics,
University of Texas Southwestern Medical Center, Dallas, Texas, United
States of America
\\
\bf{3} Division of Nutritional Sciences, Cornell University, Ithaca,
New York, United States of America
\\
\bf{4} Tri-Institutional Training Program in Computational Biology and
Medicine, New York, New York, United States of America
\\
$\ast$ Contributed equally. 
\\
$\dag$ E-mail: zg27@cornell.edu
\end{flushleft}

% Please keep the abstract between 250 and 300 words
\section*{Abstract}

Epistasis describes the phenomenon that mutations at different loci do
not have independent effects with regard to certain
phenotypes. Understanding the global epistatic landscape is vital for
many genetic and evolutionary theories. Current knowledge for
epistatic dynamics under multiple conditions is limited by the
technological difficulties in experimentally screening epistatic
relations among genes. We explored this issue by applying flux balance
analysis to simulate epistatic landscapes under various
environmental perturbations. Specifically, we looked at gene-gene
epistatic interactions, where the mutations were assumed to occur in
different genes. We predicted that epistasis tends to become more
positive from glucose-abundant to nutrient-limiting conditions,
indicating that selection might be less effective in removing
deleterious mutations in the latter. We also observed a stable core of
epistatic interactions in all tested conditions, as well as many
epistatic interactions unique to each condition. Interestingly, genes
in the stable epistatic interaction network are directly linked to
most other genes whereas genes with condition-specific epistasis form
a scale-free network. Furthermore, genes with stable epistasis tend to
have similar evolutionary rates, whereas this co-evolving relationship
does not hold for genes with condition-specific epistasis. Our
findings provide a novel genome-wide picture about epistatic dynamics
under environmental perturbations.

\section*{Introduction}


Epistasis refers to the phenomenon wherein mutations of two genes can
modify each other's phenotypic outcomes. It can be positive
(alleviating), or negative (aggravating), when a combination of
deleterious mutations shows a fitness value that is higher, or lower,
than expectation, respectively. For example, a mutation that hampers a
pathway's function may allow for other mutations in the same pathway
without a fitness consequence, resulting in positive
epistasis. Conversely, genes or pathways with redundant functions can
give rise to negative epistasis. It is well established that epistasis
is important for the evolution of sex \citep{Kondrashov1982,
Azevedo2006, Otto2007}, speciation \citep{Presgraves2007}, mutational
load \citep{Hansen2001}, ploidy \citep{Musso2008}, genetic
architecture of growth traits \citep{Xu2011}, genetic drift
\citep{Perez-Figueroa2009}, genomic complexity \citep{Sanjuan2008},
and drug resistance \citep{Trindade2009}. As biological systems in
nature have to face multiple genetic
and environmental perturbations, understanding the global landscape
and dynamics of epistasis under these perturbations remains an
important issue in the evolutionary field. In an earlier study, we
addressed genome-wide epistasis dynamics under various genetic
perturbations \citep{Xu2012}. In this study, we will investigate the impact of
environmental perturbations on global epistasis dynamics.

How epistatic interactions among genes change in different
environments has been intensively studied in various model
organisms, including \textit{E. coli} \citep{Remold2001,
Kishony2003, Cooper2005}, \textit{S. cerevisiae} \citep{Korona1999,
Szafraniec2001, Jasnos2008}, \textit{C. elegans}
\citep{Vassilieva2000, Baer2006} and \textit{D. melanogaster}
\citep{Yang2001, Fry2002, AletheaD.Wang2009}. The results of these
studies, however, are very controversial. While some studies observed
increasing positive epistasis under harsh conditions
\citep{Kishony2003, Jasnos2008, Yang2001}, others have opposite
findings \citep{Cooper2005, Korona1999, Szafraniec2001, Vassilieva2000,
Baer2006, Fry2002, AletheaD.Wang2009, Young2009}. Even within the same
species, different experimental studies might have conflicting
conclusions (e.g. \citet{Kishony2003, Cooper2005}). One possible
reason for the above controversy could have
originated from the fact that most studies only looked at the
epistasis dynamics based on a small number of genes, where the
properties cannot be generalized to the entire organism.

The main obstacle to exploring global epistatic dynamics under a
variety of environments is the difficulty of applying high-throughput
experimental platforms. To explore epistasis on a genomic scale, a
number of technologies have been developed to systematically map
genetic interaction networks, such as synthetic genetic array (SGA)
\citep{Tong2004, Costanzo2010}, diploid-based synthetic lethality
analysis with microarrays (dSLAM) \citep{Pan2004, Pan2006}, synthetic
dosage-suppression and lethality screen \citep{Measday2002,
Measday2005, Sopko2006} and epistatic miniarray profiles (EMAP)
\citep{Collins2007, Fiedler2009, Kornmann2009}. A key issue for all
these experimental studies is that these epistatic networks have been
constructed only under normal laboratory
conditions. However, cells in nature are constantly bombarded by various
external environmental stresses. Epistasis dynamics under these
perturbations cannot be predicted based on a single laboratory
condition. Few studies have constructed epistatic networks for
multiple environments. A recent study that has only
constructed epistatic networks for a group of genes with specific
functions under one normal and one harsh condition already requires a
large amount of effort \citep{Bandyopadhyay2011}. Consequently,
genome-scale epistasis landscapes under a variety of environmental
perturbations remain largely uncharacterized.

Here we explored this issue by using Flux Balance Analysis (FBA) to
simulate epistasis dynamics among genes under multiple environmental
perturbations. FBA can provide reliable predictions by 
optimizating a presumed objective function,
commonly growth maximization in microbes, subject to the known reactions and
constraints of a metabolic network \citep{Edwards2001, Shlomi2005, Becker2007, Feist2008,
Smallbone2009a, Orth2010}. Using this platform, a previous study has
investigated synthetic lethal interactions (one type of negative
epistasis) under multiple environmental perturbations and showed the
plasticity of epistatic interactions in the metabolic networks
\citep{Harrison2007}. Here we examined both positive and negative
epistasis using FBA,
and were able to show that, on a genome scale, epistatic interactions
tend to become more positive in nutrient-limiting conditions relative
to abundant-glucose media. In addition, while a large proportion of
epistatic interactions can be rewired dynamically under varying
environments, there is a set of epistatic interactions that are
stable across all tested environments. We also discovered different
network and evolutionary properties for genes with stable and dynamic
epistatic interactions. Implications of our findings were discussed.


% You may title this section "Methods" or "Models". 
% "Models" is not a valid title for PLoS ONE authors. However, PLoS ONE
% authors may use "Analysis" 
\section*{Methods}

Scripts for generating and analyzing the data can be found in the
source code repository located at
\url{https://github.com/bbarker/COBRAscripts/}. Scripts and
documentation specific to this paper are located in the subdirectory
\textit{MyProjects/EnvironmentalEpistasisFBA}.

\subsection*{Flux Balance Analysis}

Flux Balance Analysis attempts to tackle issues inherent in other
methods of metabolic modeling, such as the need to measure a large
number of parameters, slow speed of simulation, and dependence on
initial conditions \citep{Orth2010, Schellenberger2011a}. 
Other than needing a fairly complete
understanding of the reactions present in an organism, the only
measurements required to perform a genome-scale metabolic simulation
are those for determining biomass constitution or a gene expression
profile \citep{Shlomi2005, Mo2009}. Strictly speaking, FBA is a particular type of
constraint based modeling (CBM). Constraint based modeling frames the
stoichiometry that describe the reactions present in an organism as a
matrix equation with indeterminates (reaction fluxes) subject to
constraints \citep{Smallbone2009a, Mo2009}. The optimization problem
is described as follows:


\begin{equation}
\centering
\begin{tabular}{rl}
maximize   & $\mathbf{c}^T\mathbf{v}$                                     \\
subject to & $\mathbf{S v} = \frac{\D{\mathbf{x}}}{\D{t}} = \mathbf{0}$   \\
           & $\mathbf{v}_{lb} \preceq \mathbf{v} \preceq \mathbf{v}_{ub}$ \\
\end{tabular}
\end{equation}

$\mathbf{S}$ is a matrix, in which rows and columns correspond to
cellular metabolites and reactions in the reconstructed network
respectively. $\mathbf{v}$ is the reaction flux with upper and lower
bounds $\mathbf{v}_{ub}$ and $\mathbf{v}_{lb}$
respectively. Multiplying the stoichiometric matrix $\mathbf{S}$ by
the flux vector v equals the concentration change over time
($\frac{\D{\mathbf{x}}}{\D{t}}$). At steady state, the flux through
each reaction is given by $\mathbf{Sv} = \mathbf{0}$. Further details
on the underlying methods can be found in the literature 
\citep{Xu2012, Smallbone2009a, He2010}.

The fluxes of mutations employed in this analysis were restricted to
be 50\% of the wild-type fluxes found for growth rate maximization by
geometric FBA \citep{He2010}. To find new conditions with a specified carbon
source or other limiting nutrient that achieves 20\% of the
high-glucose growth rate, we can solve a linear program for the
minimization of the limiting nutrient uptake while requiring the
growth rate to be equal to 20\% of the abundant-glucose
growth-rate. For maximum growth rate conditions (Table S1, \Figs\:S2
and S4), we allowed unrestricted uptake of the limiting nutrient to
obtain the maximum growth in that condition, up to the point where it
would reach the high-glucose growth rate. Mutations affecting protein
complexes and pleiotropic genes are handled by uniform restriction
across enzymes as described before \citep{Xu2012}.

\subsection*{Definition of epistasis}

In each gene mutant pair, the epistasis value is calculated based on
the equation: $\epsilon = W_{xy} - W_xW_y$, in which $W_{xy}$ is the
fitness of an organism with two mutations in genes X and Y, whereas
$W_{x}$ or $W_{y}$ refers to the fitness of the organism with mutation
only at gene X or Y respectively. Each fitness listed previously is
calculated relative to the wild-type fitness. Absolute fitness values
are determined by the value of the biomass maximization objective
present in the model. Finally, a confidence threshold
($\left|\epsilon\right| \geq 0.01$) was applied to generate epistatic
interactions \citep{Xu2012, Costanzo2010, He2010}. We also conducted
analyses based on a different threshold for epistasis and the general
conclusions still hold in our analysis (\Figs\:S1 and S3).

\subsection*{Evolutionary rates and network parameters}

Evolutionary rates of \textit{S. cerevisiae} genes were downloaded
from supplementary materials of \citet{Wall2005}, in which orthologs were
defined by four complete genomes of \textit{Saccharomyces} species
(\textit{Saccharomyces cerevisiae}, \textit{Saccharomyces paradoxus},
\textit{Saccharomyces mikatae} and \textit{Saccharomyces bayanus}) and
evolutionary rates at synonymous and nonsynonymous sites were
calculated based on a four-way yeast species alignment for
\textit{S. cerevisiae} genes by PAML. For the distributions in
\ref{fig:eef5}A, we randomly sampled gene pairs with the same number
of gene pairs
as in three epistasis networks (epistasis in all 16 conditions,
extremely stable epistasis, and extremely dynamic epistasis,
respectively), and calculated the average evolutionary rate
differences between random gene pairs in each of these three sample
sets. The simulations were repeated 10,000 times for each of the three
groups, which are color coded to correspond to the epistasis networks
of the same size.

Network parameters such as the shortest path length, clustering
coefficient and closeness were calculated using the computer software
Pajek, downloaded from:
\url{http://vlado.fmf.uni-lj.si/pub/networks/pajek}. The shortest path
length between two genes in a network reflects the overall network
interconnectedness; the smaller the average shortest path length is,
the higher chance that genes in this network could interact with the
other genes. The clustering coefficient of a network is a measurement
of the degree to which nodes in a network tend to cluster together;
the larger the average clustering coefficient is, the more closely the
genes are connected, forming modules. The closeness of a network
measures the centrality of nodes within a network; nodes that occur on
shortest paths with other nodes have higher closeness than those that
do not \citep{Barabasi2004}.

% Results and Discussion can be combined.
\section*{Results}

% We only support three levels of headings, please do not create a heading level below \subsubsection.

\subsection*{FBA modeling and simulated growth conditions}

We applied the yeast \textit{S. cerevisiae} metabolic reconstruction
iMM904 \citep{Mo2009} to examine the dynamics of epistasis under various
environmental perturbations. The reconstruction has 904 metabolic
genes that are associated with 1,412 metabolic reactions. We conducted
FBA simulations under an abundant-glucose condition and 16
nutrient-limiting conditions. In 15 of these conditions, the carbon source
(abundant glucose) was replaced by one of the following: acetaldehyde,
acetate, adenosine 3',5'-bisphosphate, adenosyl methionine, adenosine,
alanine, allantoin, arginine, ethanol, glutamate, glutamine, glycerol,
low glucose, trehalose, and xanthosine, respectively. These conditions
represent a wide variety of nutrient and energy sources: nucleosides,
amino acids, sugars, alcohols, etc. Additionally, we looked at
abundant glucose under limited phosphorus availability.

To ensure that all environments have the same growth
rates in the following analyses, we restricted the carbon source or
phosphorous uptake levels for each of the 16 environmental
perturbations such that only 20\% of the high-glucose growth rate was
attained. This was chosen because it has been shown that metabolism
is directly linked to growth and similar growth rates often induce
similar metabolic pathways \citep{Jakubowska2012}. It is therefore
important to use a fixed growth rate among different conditions to
control for the relationship between growth rates and the overall
metabolic activity so as not to induce a growth-rate specific
effect. The 20\% high-glucose level was chosen because some media
types do not support high growth rates, regardless of the abundance of
the nutrient source. Specifically, the acetate condition had the
lowest wild-type growth rate (38.5\% of the wild-type glucose growth
rate) when unrestricted carbon uptake was permitted. In order to allow
flexibility with adding more conditions in future studies while
simultaneously not allowing extremely low growth rates that may be
possible to model but are unlikely to persist in natural environments,
we chose a growth rate of approximately half of this minimal wild-type
growth rate as the the wild-type growth rate for all conditions.

In order to estimate epistasis between genes, we
created a mutation for each gene in each condition that restricted the
flux to be 50\% of the wild-type flux found by geometric FBA for all
reactions associated with the mutant gene \citep{Xu2012}. Epistatic relations
between any two genes were calculated under each condition. We also
tested our core findings allowing maximum growth in each condition
(Table S1) and the general trends in our results remained similar, as
described in the following.

\subsection*{More positive differential epistases from rich media to
nutrient-limiting conditions}

To directly address how the sign and magnitude of epistases change
under nutrient-limiting conditions, we calculated differential
epistasis ($\D\epsilon$), which is defined as the epistatic change from
abundant-glucose media to the nutrient-limiting condition for each
gene pair in each growth condition. A gene pair with positive (or
negative) differential interaction under an environmental perturbation
is defined as the gene pair having increasing (or decreasing)
epistasis values from the abundant-glucose media to that
condition. \Fig~\ref{fig:eef1}A depicts the
distribution of differential
epistases in two growth conditions (ethanol and glycerol) as an
example. Only genes with $\left|\D\epsilon\right| \geq 0.01$ in at least
one of the two conditions are included in this figure. As quantified
in Table S2, there are 6.1\% and 5.5\% of all gene pairs with
$\left|\D\epsilon\right| \geq 0.01$ from abundant-glucose media to ethanol
and glycerol growth conditions, respectively. Among them, a large
number of gene pairs even change their sign of epistasis (Table
S2). Simulations in other conditions show similar effects (Table S2),
indicating that epistatic relationships among genes can be very
dynamic between abundant-glucose media and nutrient-limiting
conditions.

% \ifthenelse{\boolean{thesisStyle}}{% 
% \captionpage{figure}{%
% \eeFBAfigOneCap}
% \label{fig:eef1}

% \begin{figure}[H]
% \centering
% \includegraphics[height=0.95\textheight]{envFigure_1}
% \end{figure}
% }{}

We further investigated the sign of differential epistasis from
abundant-glucose to nutrient-limiting conditions. As shown in \Fig~\ref{fig:eef1}A, 
we observed more yellow dots (positive differential epistasis)
than blue dots (negative differential epistasis) in both
panels. Indeed, as quantified in \Fig~\ref{fig:eef1}B, 72\% and 57\% of
differential epistases are positive in ethanol and glycerol
conditions, respectively. We further explored all 16 nutrient-limiting
conditions and the results are shown in \Fig~\ref{fig:eef1}C. In most of our
simulated conditions (13/16), there are significantly more positive
differential epistases than negative differential epistases (Binomial
test, P $< 10^{-5}$ for each of the 13 conditions), indicating that
epistasis tends to become more positive in nutrient-limiting
conditions. This conclusion does not depend on the criteria we used to
define differential epistasis (\Fig~S1).

A recent high-throughput experiment measured epistatic relations
between roughly 80,000 gene pairs with and without perturbation by a
DNA-damaging agent (methyl methanesulfonate, MMS). The study
represents the most comprehensive experimental study so far to explore
epistatic dynamics from a rich medium to a harsh condition
\citep{Bandyopadhyay2011}. Interestingly, the authors also found more
positive differential
epistases than negative differential epistases, which is consistent
with our general observation (\Fig~\ref{fig:eef1}C). We further allowed maximum
growth in each condition and the general trends in our results
remained similar (\Fig~S2).

We found that differential epistasis had functional importance after
performing both Gene Ontology (GO) and Kyoto Encyclopedia of Genes and
Genomes (KEGG) enrichment analyses to compare genes with positive and
negative differential epistases through the glucose-abundant to
ethanol transition. We chose the ethanol condition as an example
because it is one of the most widely used conditions for the baker's
yeast. We observed that 38 GO terms and 8 KEGG pathways
are enriched for positive differential epistasis, while 18 GO terms
and 1 KEGG pathways are enriched for negative differential epistasis
(Table S3). More importantly, we found positive and negative
differential epistases uniquely contribute to different aspects of
ethanol and energy metabolism. For example, positive differential
epistasis is enriched in monohydric alcohol metabolic processes,
oxidoreductase activity acting on aldehyde group donors, the TCA
cycle, and pyruvate metabolism, while negative differential epistasis
is enriched in ethanol metabolic processes and various amino acid
terms and pathways, indicating the functional importance of
differential epistasis (Table S3). This is consistent with
experimental results that show differential epistatic interactions,
rather than static epistatic interactions, are functionally related to
the response of interest \citep{Bandyopadhyay2011}.

Several system properties were found to correlate with the ratio of
the number of positive to negative differential epistases (Table
S4). A strong correlation exists between the number of essential genes
in a given condition and the ratio of positive to negative
differential epistasis on transition from high glucose to that
condition ($\rho = 0.8392$, P $=4.8107 \times 10^{-5}$), which was a better predictor
than the number of non-zero fluxes in the wild-type vector for that
environment ($\rho = 0.5115$, P = 0.0428). Another strong predictor for
positive differential epistasis was the mean relative fitness of
single mutants in the new environment ($\rho = -0.8029$, P $= 2.7427 \times 10^{-4}$);
this anticorrelation suggests that a propensity for a lower single
mutant fitness can cause a shift towards positive epistasis.

\subsection*{Dynamic epistasis between nutrient-limiting conditions}

\Fig~\ref{fig:eef1} explored the epistasis dynamics from abundant-glucose media
to nutrient-limiting conditions. As biological systems in nature
constantly face perturbations to the environment, it is
interesting to investigate the epistasis dynamics among
nutrient-limiting conditions. To achieve this aim, we first explored
the epistatic relationship between the same gene pairs in ethanol and
glycerol environments. \Fig~\ref{fig:eef2}A lists the number of gene pairs that have various
epistatic relationships. It is noteworthy to point out that,
consistent with previously published results, there are significantly
more positive epistases than negative epistases between genes in either
condition \citep{He2010}.

% \ifthenelse{\boolean{thesisStyle}}{% 
% \begin{figure}[H]
% \centering
% \includegraphics[width=\textwidth]{envFigure_2}
% \caption{\eeFBAfigTwoCap}
% \label{fig:eef2}
% \end{figure}
% }{}

If two genes have the same sign of epistasis and
$\left|\epsilon\right| \geq 0.01$ in both conditions, they are defined
as having a similar epistatic
relationship in these two conditions. To quantify epistatic dynamics
between ethanol and glycerol growth conditions, we defined the
percentage of gene pairs with similar epistatic relations to be the
number of gene pairs with similar epistasis relations shared in these
two conditions (overlap) divided by the number of gene pairs with
epistasis in either condition (union). Our results show that 79\% of
gene pairs have similar epistasis relations between these two
conditions. \Fig~\ref{fig:eef2}B shows the distribution for the percentages of
gene pairs with similar epistasis relations between any 2 of 16
conditions, demonstrating a variable degree of epistatic similarity
between any two conditions.  This conclusion still holds when we used
different criteria to define epistatic relationships between genes
(\Fig~S3).

To understand the global distribution of all epistatic relations, we
considered 16 conditions together and calculated the fraction of
epistatic interactions existing in 1, 2, 3, \ldots, 15, and 16
conditions, respectively. As shown in \Fig~\ref{fig:eef3}A, we found that there is a
U frequency distribution for the number of growth conditions in which
a specific epistatic interaction is observed. This means that
approximately 52\% of these interactions are either condition-specific
(24\%; termed dynamic) or predicted to exist in all conditions (28\%;
termed stable), and about 48\% is intermediate (exists in multiple but not
all 16 conditions). An analogous result was obtained previously, but
only for synthetic lethal interactions \citep{Harrison2007}. We also changed the
growth assumption and allowed maximum growth in each condition and
reanalyzed the global distribution of all epistatic relations. The U
frequency distribution for the number of growth conditions in which a
specific epistatic interaction is observed remained similar
(\Fig~S4). Based on the result in \Fig~\ref{fig:eef3}A,
we further calculated the ratio of these three types of epistatic
relations in each of the 16 environmental perturbations. As shown in
\Fig~\ref{fig:eef3}B, we found that in
each environment, about 40-60\% of epistatic interactions are stable
and that each environment also has many private epistases among genes.

% \ifthenelse{\boolean{thesisStyle}}{% 
% \captionpage{figure}{%
% \eeFBAfigThreeCap}
% \label{fig:eef3}

% \begin{figure}[H]
% \centering
% \includegraphics[width=\textwidth]{envFigure_3}
% \end{figure}
% }{}

\subsection*{Different network properties for stable and dynamic epistasis}

Analysis on network properties can reveal various organization
principles (e.g. frequency of occurrence, centrality) for epistasis
networks \citep{Tong2004, Costanzo2010} and therefore provide valuable
information to further
distinguish stable and dynamic epistasis. To achieve this aim, we
compared networks formed by extremely stable and extremely dynamic
epistasis among genes and asked whether they have distinct network
properties. The degree distributions for both types of epistasis are
shown in \Fig~\ref{fig:eef4}A. Interestingly, extremely stable epistatic
interactions form an exponential network architecture, which is
homogeneous, meaning that most nodes have a very similar number of
links (\Fig~\ref{fig:eef4}A, left panel). In contrast, the extremely dynamic
epistatic interactions give rise to a scale-free network topology,
which is heterogeneous, meaning that the majority of nodes have few
links but a small number of hubs have a large number of links
(\Fig~\ref{fig:eef4}A, right panel).

% \ifthenelse{\boolean{thesisStyle}}{% 
% \begin{figure}[H]
% \centering
% \includegraphics[width=\textwidth]{envFigure_4}
% \caption{\eeFBAfigFourCap}
% \label{fig:eef4}
% \end{figure}
% }{}

In addition, we calculated three network parameters to compare these
two types of epistatic interactions. We found that the network formed
by extremely stable epistases has a smaller shortest path length, a
larger clustering coefficient and larger closeness than the network
formed by extremely dynamic epistases (\Fig~\ref{fig:eef4}B). These results are
consistent with the scenario that genes with extremely stable
epistasis are directly linked to most other genes and form an
exponential network topology, while genes with extremely dynamic
epistasis form a scale-free network. Our results also show that the
network induced by intermediate epistases have intermediate values of
these parameters compared to that of extremely stable and extremely
dynamic epistasis networks (Tables S5 and S6).

\subsection*{Co-evolution of genes with epistatic interaction}

Gene pairs with epistasis identified in real experiments usually show
similar evolutionary rates. To investigate whether two genes with
predicted epistasis also tend to co-evolve, we calculated the
evolutionary rate differences between two genes with epistasis from
FBA modeling (\Fig~\ref{fig:eef5}A). Evolutionary rates
(${d_\textnormal{N}}$/${d_\textnormal{S}}$) based on orthologous gene
sets
from four yeast species of the genus Saccharomyces were downloaded
from a commonly used reference dataset \citep{Wall2005}. Simulations
based on the same number of gene pairs with FBA-predicted epistasis
were conducted to estimate the evolutionary rate differences for any
two randomly selected genes. As shown in \Fig~\ref{fig:eef5}A, the gene pairs
with FBA-predicted epistatic interactions tend to have more similar
evolutionary rates than random expectation (P $< 10^{-4}$).

% \ifthenelse{\boolean{thesisStyle}}{% 
% \captionpage{figure}{%
% \eeFBAfigFiveCap}
% \label{fig:eef5}

% \begin{figure}[H]
% \centering
% \includegraphics[height=0.95\textheight]{envFigure_5}
% \end{figure}
% }{}

In \Fig~\ref{fig:eef4} we observed unique network properties for extremely stable
and extremely dynamic epistatic interactions. We further investigate
the co-evolution between genes with these two types of epistatic
relationships. As shown in \Fig~\ref{fig:eef5}A, genes with extremely stable
epistasis tend to co-evolve (P $< 10^{-4}$), while the difference between
genes with extremely dynamic epistasis and random expectation becomes
much smaller (P = 0.06). The evolutionary rate difference between gene
pairs with extremely stable and extremely dynamic epistasis is also
significant (\textit{t}-test, P $= 8\times 10^{-6}$). This difference is not caused by
genes that are involved in extremely stable or extremely dynamic
epistasis, because these two groups of genes do not have significantly
different evolutionary rates (\textit{t}-test, P = 0.796, \Fig~\ref{fig:eef5}B).

\section*{Discussion}


\subsection*{Natural selection in nutrient-limiting conditions}

Whether a genetic mutation has a fitness consequence depends on other
sites, a phenomenon called epistasis (see \citet{Lehner2011} for a
recent review on
molecular mechanisms). Positive epistasis alleviates the total harm
when multiple deleterious mutations combine together and thus reduces
the effectiveness of natural selection in removing these deleterious
mutations, whereas negative epistasis plays the opposite role by
increasing the efficiency of purging deleterious mutations by natural
selection. Results from this study present an initial glimpse over
environment-induced epistasis dynamics at the genome scale. Using
differential epistasis from abundant-glucose to nutrient-limiting
conditions, our results show that epistasis between specific genes can
become more positive or more negative in nutrient-limiting conditions,
which is consistent with previous findings in small scale studies
\citep{Remold2001, Kishony2003, Cooper2005, Korona1999, Szafraniec2001,
Jasnos2008, Vassilieva2000, Baer2006, Yang2001, Fry2002,
AletheaD.Wang2009, Young2009}. However, we showed that, at the genome
scale, epistasis is
more positive in nutrient-limiting conditions. Interestingly, our
simulation results are consistent with a recent genome-wide study
between laboratory and harsh growth conditions
\citep{Bandyopadhyay2011}. How epistasis affects selection in harsh
conditions has been controversial \citep{Agrawal2010}. Our
results provide the genome-wide evidence arguing that selection might
be less effective in removing deleterious mutations in harsh
conditions, which could be one of the underlying reasons for a recent
observation that stimulation of a stress response can reduce mutation
penetrance in \textit{Caenorhabditis elegans} \citep{Casanueva2012}.

\subsection*{Network properties and evolutionary patterns for stable
and dynamic epistasis}

Our results indicate that epistasis could be extremely stable or
dynamic among various environmental perturbations, which is consistent
with a previous FBA study investigating synthetic lethal relations
among non-essential genes \citep{Harrison2007}. The inclusion of
essential genes in our study allows for investigation on many
important metabolic
pathways that were not previously analyzed. Nevertheless, the
distribution of epistasis among multiple environments (\Fig~\ref{fig:eef3}A)
remains largely unchanged from the previous study \citep{Harrison2007}
even when essential genes are included.

We also found that stable and dynamic epistatic relationships show
totally different network properties and evolutionary patterns, which
might provide new biological and evolutionary insights. The gene pairs
with stable epistases tend to co-evolve with each other. In addition,
from the biological pathway perspective, the smaller shortest path
length and larger closeness values in the stable epistasis network
both imply that genes with stable epistases tend to be functionally
associated with a large number of neighbors to form a condensed
functional network, different from genes in the dynamic epistasis
network that are loosely connected. Furthermore, the large clustering
coefficient in the stable epistasis network also supports the idea
that genes with stable epistasis interactions form a network core in
the whole epistasis network. Combined with observations in \Fig~\ref{fig:eef5},
this core module of epistasis in the metabolic network might represent
stable functional associations between genes that are essential for
important biological functions and evolutionarily conserved even under
different environmental perturbations. The lack of co-evolutionary
pattern and scale-free network properties for the dynamic epistasis
network, however, might represent unstable functional associations
between genes, which may only be responsible for unique functions
under specific conditions.

\subsection*{Implications and significance for exploring stable and
dynamic epistasis}

Our prediction about stable and dynamic epistasis could have important
functional applications. A recent study showed that the synthetic
lethal (negative epistasis) relationship between fumarate hydratase
and haem oxygenase can be employed successfully to identify an in
vitro drug target in hereditary leiomyomatosis and renal-cell cancer
(HLRCC) cells \citep{Frezza2011}. Exploring both dynamic and stable
epistasis could be useful in this context; stable epistatic
interactions may be
important for drug target detection in cancer or other pathogens,
whereas it may sometimes be necessary to exploit dynamic epistatic
relationships, possibly induced by treatment with an external
perturbation.

Furthermore, rational evolutionary design techniques such as OptKnock
\citep{Burgard2003} and OptGene \citep{Patil2005} attempt to find
which knockouts will enable a reaction of interest to be coupled with
growth (i.e. have positive
epistasis with growth-associated genes in a specific
environment). However, these techniques do not take into account
epistasis dynamics across different environments. In this study, we
have found that epistatic relations can be highly dynamic under
various environmental perturbations, which raises the possibility to
improve these techniques by considering epistasis dynamics in future
studies. Research on using compensatory perturbations to reach desired
network states is ongoing \citep{Cornelius2011}.

\subsection*{Caveats and future directions} Though we show several
novel insights into how varying environments can influence epistasis,
several caveats should be addressed. First, the FBA modeling used in
this study, which was proven to have great predictive power and has
been successfully employed in addressing numerous research problems
\citep{Edwards2001, Shlomi2005, Smallbone2009a}, 
only includes metabolic genes. Second, even though FBA
offers the most comprehensive simulation method for studying
epistasis, there are many improvements that can be made in order to
capture the empirically observed set of epistatic interactions
\citep{Covert2001}. For example, integrating transcriptional
regulation and physical interactions into this framework could improve
the current methods in predicting epistasis and other evolutionary
processes \citep{Szappanos2011}. Related to
this point, FBA as used herein only considers the steady state and
does not take into account any dynamics or initial conditions, and
would necessarily miss any epistatic interactions that are due to
dynamics in the system, such as changing concentrations; dynamic FBA
(which is part of rFBA) might be a solution, but would likely require
about a minimum of two orders of magnitude increase in computation
time \citep{Covert2001, Varma1994}. Recent work
on new objective functions targeting metabolite turnover rather than
flux per se has also proven successful in recovering many epistases
that were previously not found with FBA \citep{Brochado2012}.

Third, in order to understand the impact of environmental
perturbations on epistasis, we used a reductive approach and only
considered one mutation type per gene to simulate the global epistatic
landscape in 16 environments. There are countless environments in
nature. Furthermore, different mutations in the same gene and the
interactions between genes and environment can likely have an even
more complex impact on the epistasis dynamics. While it would be ideal
to simulate a larger variety of environments for multiple
mutations of the same gene, the computational cost is a limiting
factor. Our previous study showed that different mutants of the same
gene could have very dynamic epistatic interaction partners in a
single environment \citep{Xu2012}. In this study, we chose to use one
mutation per gene as we are focusing on addressing how different
environments could affect gene epistasis dynamics. Nevertheless, in
order to see how sensitive our results were, we performed the analysis
for our core results by simulating 16 environments using different
growth assumption, where the organisms are allowed to have
unrestricted uptake of the limiting nutrient to obtain the maximum
growth in that condition. We found the major trends in our results are
largely unchanged (\Figs\:S2 and S4; Table S1).

Keeping these issues in mind, our analysis uncovered several prominent
features of epistatic interactions under a variety of environmental
perturbations, and call on future effort to confirm these simulation
results using high-throughput experimental platforms. More
importantly, the enrichment of stable and dynamic epistasis provides a
new perspective to understand how biological systems may rewire
epistasis in nature.

% Do NOT remove this, even if you are not including acknowledgments.

\section*{Acknowledgments}

We firstly thank Tim Connallon for extremely helpful discussion
related to epistasis. Discussions from Dr. Huifeng Jiang, Dr. Chris
Myers, and Mr. Kaixiong Ye are also very much appreciated.

%\section*{References}

% Either type in your references using
% \begin{thebibliography}{}
% \bibitem{}
% Text
% \end{thebibliography}
%
% OR
%
% Compile your BiBTeX database using our plos2009.bst
% style file and paste the contents of your .bbl file
% here.
% 

\begin{thebibliography}{10}
\providecommand{\url}[1]{\texttt{#1}}
\providecommand{\urlprefix}{URL }
\expandafter\ifx\csname urlstyle\endcsname\relax
  \providecommand{\doi}[1]{doi:\discretionary{}{}{}#1}\else
  \providecommand{\doi}{doi:\discretionary{}{}{}\begingroup
  \urlstyle{rm}\Url}\fi
\providecommand{\bibAnnoteFile}[1]{%
  \IfFileExists{#1}{\begin{quotation}\noindent\textsc{Key:} #1\\
  \textsc{Annotation:}\ \input{#1}\end{quotation}}{}}
\providecommand{\bibAnnote}[2]{%
  \begin{quotation}\noindent\textsc{Key:} #1\\
  \textsc{Annotation:}\ #2\end{quotation}}
\providecommand{\eprint}[2][]{\url{#2}}

\bibitem{Kondrashov1982}
Kondrashov AS (1982) {Selection against harmful mutations in large sexual and
  asexual populations}.
\newblock Genetics Research 40: 325--332.

\bibitem{Azevedo2006}
Azevedo RBR, Lohaus R, Srinivasan S, Dang KK, Burch CL (2006) {Sexual
  reproduction selects for robustness and negative epistasis in artificial gene
  networks.}
\newblock Nature 440: 87--90.

\bibitem{Otto2007}
Otto SP (2007) {Unravelling the evolutionary advantage of sex: a commentary on
  'Mutation-selection balance and the evolutionary advantage of sex and
  recombination' by Brian Charlesworth.}
\newblock Genetical research 89: 447--9.

\bibitem{Presgraves2007}
Presgraves DC (2007) {Speciation Genetics: Epistasis, Conflict and the Origin
  of Species}.
\newblock Current Biology 17: R125--R127.

\bibitem{Hansen2001}
Hansen TF, Wagner GP (2001) {Epistasis and the Mutation Load: A
  Measurement-Theoretical Approach}.
\newblock Genetics 158: 477--485.

\bibitem{Musso2008}
Musso G, Costanzo M, Huangfu M, Smith AM, Paw J, et~al. (2008) {The extensive
  and condition-dependent nature of epistasis among whole-genome duplicates in
  yeast}.
\newblock Genome Research 18: 1092--1099.

\bibitem{Xu2011}
Xu L, Jiang H, Chen H, Gu Z (2011) {Genetic Architecture of Growth Traits
  Revealed by Global Epistatic Interactions}.
\newblock Genome Biology and Evolution 3: 909--914.

\bibitem{Perez-Figueroa2009}
P\'{e}rez-Figueroa A, Caballero A, Garc\'{\i}a-Dorado A, L\'{o}pez-Fanjul C
  (2009) {The Action of Purifying Selection, Mutation and Drift on Fitness
  Epistatic Systems}.
\newblock Genetics 183: 299--313.

\bibitem{Sanjuan2008}
Sanju\'{a}n R, Nebot MR (2008) {A network model for the correlation between
  epistasis and genomic complexity}.
\newblock PLoS ONE 3.

\bibitem{Trindade2009}
Trindade S, Sousa A, Xavier KB, Dionisio F, Ferreira MG, et~al. (2009)
  {Positive Epistasis Drives the Acquisition of Multidrug Resistance}.
\newblock PLoS Genet 5: e1000578.

\bibitem{Xu2012}
Xu L, Barker B, Gu Z (2012) {Dynamic epistasis for different alleles of the
  same gene.}
\newblock Proceedings of the National Academy of Sciences of the United States
  of America 109: 10420--10425.

\bibitem{Remold2001}
Remold SK, Lenski RE (2001) {Contribution of individual random mutations to
  genotype-by-environment interactions in Escherichia coli}.
\newblock Proceedings of the National Academy of Sciences 98: 11388--11393.

\bibitem{Kishony2003}
Kishony R, Leibler S (2003) {Environmental stresses can alleviate the average
  deleterious effect of mutations}.
\newblock Journal of Biology 2: 14.

\bibitem{Cooper2005}
Cooper TF, Lenski RE, Elena SF (2005) {Parasites and mutational load: an
  experimental test of a pluralistic theory for the evolution of sex.}
\newblock Proceedings Biological sciences / The Royal Society 272: 311--317.

\bibitem{Korona1999}
Korona R (1999) {Genetic Load of the Yeast Saccharomyces cerevisiae under
  Diverse Environmental Conditions}.
\newblock Evolution 53: 1966--1971.

\bibitem{Szafraniec2001}
Szafraniec K, Borts RH, Korona R (2001) {Environmental stress and mutational
  load in diploid strains of the yeast Saccharomyces cerevisiae}.
\newblock Proceedings of the National Academy of Sciences 98: 1107--1112.

\bibitem{Jasnos2008}
Jasnos L, Tomala K, Paczesniak D, Korona R (2008) {Interactions Between
  Stressful Environment and Gene Deletions Alleviate the Expected Average Loss
  of Fitness in Yeast}.
\newblock Genetics 178: 2105--2111.

\bibitem{Vassilieva2000}
Vassilieva LL, Hook AM, Lynch M (2000) {THE FITNESS EFFECTS OF SPONTANEOUS
  MUTATIONS IN CAENORHABDITIS ELEGANS}.
\newblock Evolution 54: 1234--1246.

\bibitem{Baer2006}
Baer CF, Phillips N, Ostrow D, Avalos A, Blanton D, et~al. (2006) {Cumulative
  Effects of Spontaneous Mutations for Fitness in Caenorhabditis: Role of
  Genotype, Environment and Stress}.
\newblock Genetics 174: 1387--1395.

\bibitem{Yang2001}
Yang HP, Tanikawa AY, {Van Voorhies} WA, Silva JC, Kondrashov AS (2001)
  {Whole-Genome Effects of Ethyl Methanesulfonate-Induced Mutation on Nine
  Quantitative Traits in Outbred Drosophila melanogaster}.
\newblock Genetics 157: 1257--1265.

\bibitem{Fry2002}
Fry JD, Heinsohn SL (2002) {Environment Dependence of Mutational Parameters for
  Viability in Drosophila melanogaster}.
\newblock Genetics 161: 1155--1167.

\bibitem{AletheaD.Wang2009}
Wang AD, Sharp NP, Spencer CC, Tedman-Aucoin K, Agrawal AF (2009) {Selection,
  Epistasis, and Parent‐of‐Origin Effects on Deleterious Mutations across
  Environments in Drosophila melanogaster.}
\newblock The American Naturalist 174: 863--874.

\bibitem{Young2009}
Young JA, Yourth CP, Agrawal AF (2009) {The effect of pathogens on selection
  against deleterious mutations in Drosophila melanogaster}.
\newblock Journal of Evolutionary Biology 22: 2125--2129.

\bibitem{Tong2004}
Tong AHY, Lesage G, Bader GD, Ding H, Xu H, et~al. (2004) {Global Mapping of
  the Yeast Genetic Interaction Network}.
\newblock Science 303: 808--813.

\bibitem{Costanzo2010}
Costanzo M, Baryshnikova A, Bellay J, Kim Y, Spear ED, et~al. (2010) {The
  genetic landscape of a cell.}
\newblock Science (New York, NY) 327: 425--31.

\bibitem{Pan2004}
Pan X, Yuan DS, Xiang D, Wang X, Sookhai-Mahadeo S, et~al. (2004) {A robust
  toolkit for functional profiling of the yeast genome}.
\newblock Molecular Cell 16: 487--496.

\bibitem{Pan2006}
Pan X, Ye P, Yuan DS, Wang X, Bader JS, et~al. (2006) {A DNA integrity network
  in the yeast Saccharomyces cerevisiae}.
\newblock Cell 124: 1069--1081.

\bibitem{Measday2002}
Measday V, Hieter P (2002) {Synthetic dosage lethality}.
\newblock In: GUIDE TO YEAST GENETICS AND MOLECULAR AND CELL BIOLOGY, PT B, 525
  B STREET, SUITE 1900, SAN DIEGO, CA 92101-4495 USA: ACADEMIC PRESS INC,
  volume 350 of \emph{METHODS IN ENZYMOLOGY}. pp. 316--326.

\bibitem{Measday2005}
Measday V, Baetz K, Guzzo J, Yuen K, Kwok T, et~al. (2005) {Systematic yeast
  synthetic lethal and synthetic dosage lethal screens identify genes required
  for chromosome segregation.}
\newblock Proceedings of the National Academy of Sciences of the United States
  of America 102: 13956--13961.

\bibitem{Sopko2006}
Sopko R, Huang D, Preston N, Chua G, Papp B, et~al. (2006) {Mapping pathways
  and phenotypes by systematic gene overexpression}.
\newblock Molecular Cell 21: 319--330.

\bibitem{Collins2007}
Collins SR, Miller KM, Maas NL, Roguev A, Fillingham J, et~al. (2007)
  {Functional dissection of protein complexes involved in yeast chromosome
  biology using a genetic interaction map.}
\newblock Nature 446: 806--810.

\bibitem{Fiedler2009}
Fiedler D, Braberg H, Mehta M, Chechik G, Cagney G, et~al. (2009) {Functional
  organization of the S. cerevisiae phosphorylation network.}
\newblock Cell 136: 952--63.

\bibitem{Kornmann2009}
Kornmann B, Currie E, Collins SR, Schuldiner M, Nunnari J, et~al. (2009) {An
  ER-mitochondria tethering complex revealed by a synthetic biology screen.}
\newblock Science (New York, NY) 325: 477--481.

\bibitem{Bandyopadhyay2011}
Bandyopadhyay S (2011) {Rewiring of Genetic Networks in Response to DNA
  Damage}.
\newblock Science 1385.

\bibitem{Edwards2001}
Edwards JS, Ibarra RU, Palsson BO (2001) {In silico predictions of Escherichia
  coli metabolic capabilities are consistent with experimental data.}
\newblock Nature biotechnology 19: 125--130.

\bibitem{Shlomi2005}
Shlomi T, Berkman O, Ruppin E (2005) {Regulatory on/off minimization of
  metabolic flux}.
\newblock Proceedings of the National Academy of Sciences 102: 7695--7700.

\bibitem{Becker2007}
Becker SA, Feist AM, Mo ML, Hannum G, Palsson BO, et~al. (2007) {Quantitative
  prediction of cellular metabolism with constraint-based models: the COBRA
  Toolbox.}
\newblock Nature protocols 2: 727--38.

\bibitem{Feist2008}
Feist AM, Palsson BO (2008) {The growing scope of applications of genome-scale
  metabolic reconstructions using Escherichia coli.}
\newblock Nature biotechnology 26: 659--667.

\bibitem{Smallbone2009a}
Smallbone K, Simeonidis E (2009) {Flux balance analysis: a geometric
  perspective.}
\newblock Journal of theoretical biology 258: 311--5.

\bibitem{Orth2010}
Orth JD, Thiele I, Palsson BO (2010) {What is flux balance analysis?}
\newblock Nature biotechnology 28: 245--8.

\bibitem{Harrison2007}
Harrison R, Papp B, P\'{a}l C, Oliver SG, Delneri D (2007) {Plasticity of
  genetic interactions in metabolic networks of yeast.}
\newblock Proceedings of the National Academy of Sciences of the United States
  of America 104: 2307--12.

\bibitem{Schellenberger2011a}
Schellenberger J, Que R, Fleming RMT, Thiele I, Orth JD, et~al. (2011)
  {Quantitative prediction of cellular metabolism with constraint-based models:
  the COBRA Toolbox v2.0}.
\newblock Nat Protocols 6: 1290--1307.

\bibitem{Mo2009}
Mo ML, Palsson BO, Herrg\aa~rd MJ (2009) {Connecting extracellular metabolomic
  measurements to intracellular flux states in yeast}.
\newblock BMC Systems Biology 3: 37.

\bibitem{He2010}
He X, Qian W, Wang Z, Li Y, Zhang J (2010) {Prevalent positive epistasis in
  Escherichia coli and Saccharomyces cerevisiae metabolic networks.}
\newblock Nature genetics 42: 272--6.

\bibitem{Wall2005}
Wall DP, Hirsh AE, Fraser HB, Kumm J, Giaever G, et~al. (2005) {Functional
  genomic analysis of the rates of protein evolution}.
\newblock Proceedings of the National Academy of Sciences of the United States
  of America 102: 5483--5488.

\bibitem{Barabasi2004}
Barabasi AL, Oltvai ZN (2004) {Network biology: understanding the cell's
  functional organization}.
\newblock Nat Rev Genet 5: 101--113.

\bibitem{Jakubowska2012}
Jakubowska A, Korona R (2012) {Epistasis for Growth Rate and Total Metabolic
  Flux in Yeast}.
\newblock PLoS ONE 7: e33132.

\bibitem{Lehner2011}
Lehner B (2011) {Molecular mechanisms of epistasis within and between genes}.
\newblock Trends in Genetics 27: 323--331.

\bibitem{Agrawal2010}
Agrawal AF, Whitlock MC (2010) {Environmental duress and epistasis: how does
  stress affect the strength of selection on new mutations?}
\newblock Trends in ecology \& evolution 25: 450--8.

\bibitem{Casanueva2012}
Casanueva MO, Burga A, Lehner B (2012) {Fitness Trade-Offs and Environmentally
  Induced Mutation Buffering in Isogenic C. elegans}.
\newblock Science 335: 82--85.

\bibitem{Frezza2011}
Frezza C, Zheng L, Folger O, Rajagopalan KN, MacKenzie ED, et~al. (2011) {Haem
  oxygenase is synthetically lethal with the tumour suppressor fumarate
  hydratase.}
\newblock Nature 477: 225--8.

\bibitem{Burgard2003}
Burgard AP, Pharkya P, Maranas CD (2003) {Optknock: A bilevel programming
  framework for identifying gene knockout strategies for microbial strain
  optimization}.
\newblock Biotechnology and Bioengineering 84: 647--657.

\bibitem{Patil2005}
Patil K, Rocha I, Forster J, Nielsen J (2005) {Evolutionary programming as a
  platform for in silico metabolic engineering}.
\newblock BMC Bioinformatics 6: 308.

\bibitem{Cornelius2011}
Cornelius SP, Kath WL, Motter AE (2011) {Controlling Complex Networks with
  Compensatory Perturbations} .

\bibitem{Covert2001}
Covert MW, Schilling CH, Palsson BO (2001) {Regulation of gene expression in
  flux balance models of metabolism.}
\newblock Journal of theoretical biology 213: 73--88.

\bibitem{Szappanos2011}
Szappanos B, Kov\'{a}cs K, Szamecz B, Honti F, Costanzo M, et~al. (2011) {An
  integrated approach to characterize genetic interaction networks in yeast
  metabolism.}
\newblock Nature genetics 43: 656--662.

\bibitem{Varma1994}
Varma A, Palsson BO (1994) {Stoichiometric flux balance models quantitatively
  predict growth and metabolic by-product secretion in wild-type Escherichia
  coli W3110.}
\newblock Applied and environmental microbiology 60: 3724--31.

\bibitem{Brochado2012}
Brochado AR, Andrejev S, Maranas CD, Patil KR (2012) {Impact of Stoichiometry
  Representation on Simulation of Genotype-Phenotype Relationships in Metabolic
  Networks}.
\newblock PLoS Comput Biol 8: e1002758.

\end{thebibliography}

\section*{Figure Legends}
% This section is for figure legends only, do not include
% graphics in your manuscript file.
%
%\begin{figure}
%\caption{
%{\bf Bold the first sentence.}  Rest of figure caption.  
%}
%\label{Figure_label}
%\end{figure}

\begin{figure}[!htb]
\caption{
{\bf More positive differential epistases under environmental
perturbations.} (\textbf{A}) Heat maps describe the global dynamics of
differential epistasis from abundant-glucose medium to ethanol (left
panel) and glycerol (right panel) conditions. Only gene pairs with
$\left|\D\epsilon\right| \geq 0.01$ in either condition are included
in the figure. Different colors represent differential epistasis
values as indicated by the color bar at the bottom. The differential
epistasis values are assigned to be 0.1 (or -0.1) in the heat-maps
when it is greater than 0.1 (or less than -0.1). It is noteworthy to
point out that the epistasis patterns are indeed very different
between the two conditions (\Fig~\ref{fig:eef2}A). (\textbf{B})
Percentage of positive and negative differential epistases under
ethanol and glycerol conditions. (\textbf{C}) Ratio of positive to
negative differential epistases
in each simulated condition. The result from a high-throughput
experiment is also shown. The letters A-P represent acetaldehyde,
acetate, adenosine 3',5'-bisphosphate, adenosyl methionine, adenosine,
alanine, allantoin, arginine, ethanol, glutamate, glutamine, glycerol,
low glucose, phosphate, trehalose, and xanthosine, respectively.
}
\label{fig:eef1}
\end{figure}

\begin{figure}[!htb]
\caption{
{\bf Epistasis dynamics between environmental perturbations.} (\textbf{A}) 
Number of gene pairs with various epistatic relationships between ethanol and
glycerol growth conditions. (\textbf{B}) The distribution for the percentages
of gene pairs with similar epistasis relation between any 2 of 16
conditions. The frequency is derived from the 120 pairs of
environments simulated in this study.
}
\label{fig:eef2}
\end{figure}


\begin{figure}[!htb]
\caption{
{\bf The global distribution of epistatic relations under simulated
conditions.} (\textbf{A}) Distribution for the number of conditions in which
each epistatic interaction exists. Note that about 28\% of
epistatic relations are extremely stable (the very right bar) and
about 24\% are extremely dynamic (the very left bar). (\textbf{B}) Fraction
of three types of epistatic relations in each of the 16 environmental
perturbations, as indicated by the color bar to the right. The numbers
in the brackets represent the number of conditions in which each
epistatic interaction exists, as indicated in (\textbf{A}). The letters A-P
represent the simulated conditions as indicated in \Fig~\ref{fig:eef1}.
}
\label{fig:eef3}
\end{figure}

\begin{figure}[!htb]
\caption{
{\bf Network properties for the extremely stable and extremely dynamic
epistatic interactions.} (\textbf{A}) Degree distribution for genes in two
epistatic interaction networks. The networks have nodes that
correspond to genes and edges that correspond to epistatic
interactions. (\textbf{B}) Three network parameters (the definition of which
are shown in Methods) for two epistatic interaction networks.
}
\label{fig:eef4}
\end{figure}

\begin{figure}[!htb]
\caption{
{\bf Co-evolution between genes with epistasis.} (\textbf{A}) Average evolutionary
rate differences between gene pairs with FBA-predicted epistasis
(green), extremely dynamic epistasis (blue) and extremely stable
epistasis (red) are highlighted by three arrows, respectively. The
random simulations with the same number of gene pairs as each of the
three groups were repeated 10,000 times and the frequency
distributions are shown (marked by the same colors as the
corresponding arrows, respectively). (\textbf{B}) The evolutionary rates for
genes that are involved in extremely stable and extremely dynamic
epistasis, respectively. The error bars represent standard errors.
}
\label{fig:eef5}
\end{figure}


%\section*{Tables}
%
% 
% See introductory notes if you wish to include sideways tables.
%
% NOTE: Please look over our table guidelines at http://www.plosone.org/static/figureGuidelines#tables to make sure that your tables meet our requirements. Certain types of spacing, cell merging, and other formatting tricks may have unintended results and will be returned for revision.
%
%\begin{table}[!ht]
%\caption{
%\bf{Table title}}
%\begin{tabular}{|c|c|c|}
%table information
%\end{tabular}
%\begin{flushleft}Table caption
%\end{flushleft}
%\label{tab:label}
% \end{table}

\section*{Supporting Information Legends}
%
% Please enter your Supporting Information captions below in the following format:
%\item{\bf Figure SX. Enter mandatory title here.} Enter optional descriptive information here.
% 
%\begin{description}
%\item {\bf}
%\item {\bf}
%\end{description}

\begin{description}
\item {\bf Figure S1. More positive differential epistases under environmental perturbations
for different thresholds of differential epistasis}
($\left|\D\epsilon\right| \geq 0.001$, \textbf{A}) and
($\left|\D\epsilon\right| \geq 0.05$, \textbf{B}). Ratio of positive
to negative differential epistases in each simulated condition are
shown. The letters A-P represent acetaldehyde, acetate, adenosine
3',5'-bisphosphate, adenosyl methionine, adenosine, alanine,
allantoin, arginine, ethanol, glutamate, glutamine, glycerol, low
glucose, phosphate, trehalose, and xanthosine, respectively.

\item {\bf Figure S2. Analogous to \Fig~\ref{fig:eef1}B-C, but using a
maximum growth rate for each condition, where the maximum is
constrained to be no higher than the high-glucose growth rate.}
(\textbf{A}) Percentage of positive and negative differential
epistases under ethanol and glycerol conditions. (\textbf{B}) Ratio of
positive to negative differential epistases in each simulated
condition. The result from a high-throughput experiment is also
shown. The letters A-P represent acetaldehyde, acetate, adenosine
3',5'-bisphosphate, adenosyl methionine, adenosine, alanine,
allantoin, arginine, ethanol, glutamate, glutamine, glycerol, low
glucose, phosphate, trehalose, and xanthosine, respectively. Note that
in (B), low glucose has the same growth rate as high-glucose, but has
different epistatic interactions since we still use the high-oxygen
uptake level associated with the low glucose condition.

\item {\bf Figure S3. Epistasis dynamics between environmental
perturbations under different epistasis definition.} (\textbf{A})
Number of gene pairs with various epistatic relationships between
ethanol and glycerol growth conditions under a lower
($\left|\epsilon\right| \geq 0.001$) and a higher
($\left|\epsilon\right| \geq 0.05$) epistasis threshold. (\textbf{B})
The distribution for the percentages of gene pairs with similar
epistasis relations between any 2 of 16 conditions under a lower
($\left|\epsilon\right| \geq 0.001$) and a higher
($\left|\epsilon\right| \geq 0.05$) epistasis threshold.

\item {\bf Figure S4. Analogous to \Fig~\ref{fig:eef3}A, but using a
maximum growth rate for each condition, where the maximum is
constrained to be no higher than the high-glucose growth rate.}
Distribution for the number of conditions in which each epistatic
interaction exists. Note that about 26\% of epistatic relations are
extremely stable (the very right bar) and about 19\% are extremely
dynamic (the very left bar). 

\item {\bf Table S1. Wild-type growth rates used in the maximal growth
rate simulations used for \Figs S2 and S4.}

\item {\bf Table S2. Condition-specific epistases and sign-epistases
prevalence in the iMM904 yeast model.}

\item {\bf Table S3. GO term enrichment analysis results for
differential epistasis in transition to ethanol.}

\item {\bf Table S4. Properties of simulated systems that correlate
with the ratio of positive to negative differential epistases.}

\item {\bf Table S5. List of epistatic interactions for the extremely
stable, dynamic, and intermediate epistasis networks.}

\item {\bf Table S6. Table of network parameters for stable, dynamic,
and intermediate epistasis.}


\end{description}

\end{document}

