\documentclass[compress,red,notes]{beamer}
% Useful for practice: %can use [notes] or [notes=only]
% Useful for large documents with lots of images: [draft]
\usepackage{etex} %helps when using lots of packages
\mode<presentation>

\usepackage{amsmath}
\usepackage{amsthm}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
                                       %
% change this line for portable code:  %
\newcommand*{\commonDir}{../../common/}%
\input{\commonDir preambleCommon}      %
                                       %
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\usepackage{listings}
\usepackage{atslistings}
\usepackage[bigfiles]{media9}
\usepackage{stfloats}
\captionsetup[figure]{labelformat=empty}

\graphicspath{{./figures/}{../../falconBioinfo/figures/}%
{../../epistasisSameGeneArXiv/figures/}{../../thesis/figures/}}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
                              %
\input{\commonDir logos}      %
                              %
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


\usetheme{Warsaw}
% other themes: AnnArbor, Antibes, Bergen, Berkeley, Berlin, Boadilla, boxes, CambridgeUS, Copenhagen, Darmstadt, default, Dresden, Frankfurt, Goettingen,
% Hannover, Ilmenau, JuanLesPins, Luebeck, Madrid, Maloe, Marburg, Montpellier, PaloAlto, Pittsburg, Rochester, Singapore, Szeged, classic

%\usecolortheme{lily}
% color themes: albatross, beaver, beetle, crane, default, dolphin, dov, fly, lily, orchid, rose, seagull, seahorse, sidebartab, structure, whale, wolverine

%\usefonttheme{serif}
% font themes: default, professionalfonts, serif, structurebold, structureitalicserif, structuresmallcapsserif

\hypersetup{pdfpagemode=FullScreen} % makes your presentation go automatically to full screen

% define your own colors:
\definecolor{CURed}{rgb}{0.702,0.106,0.106}
\definecolor{CUGrey}{rgb}{0.302,0.31,0.3255}
%
\definecolor{White}{rgb}{1,1,1}
\definecolor{Red}{rgb}{1,0,0}
\definecolor{Blue}{rgb}{0,0,1}
\definecolor{Green}{rgb}{0,1,0}
\definecolor{Yellow}{rgb}{1,1,0}
\definecolor{magenta}{rgb}{1,0,.6}
\definecolor{lightblue}{rgb}{0,.5,1}
\definecolor{lightpurple}{rgb}{.6,.4,1}
\definecolor{gold}{rgb}{.6,.5,0}
\definecolor{orange}{rgb}{1,0.4,0}
\definecolor{hotpink}{rgb}{1,0,0.5}
\definecolor{newcolor2}{rgb}{.5,.3,.5}
\definecolor{newcolor}{rgb}{0,.3,1}
\definecolor{newcolor3}{rgb}{1,0,.35}
\definecolor{darkgreen1}{rgb}{0, .35, 0}
\definecolor{darkgreen}{rgb}{0, .6, 0}
\definecolor{darkred}{rgb}{.75,0,0}

\xdefinecolor{olive}{cmyk}{0.64,0,0.95,0.4}
\xdefinecolor{purpleish}{cmyk}{0.75,0.75,0,0}

% can also choose different themes for the "inside" and "outside"

% \usepackage{beamerinnertheme_______}
% inner themes include circles, default, inmargin, rectangles, rounded

% \usepackage{beamerouterthemesmoothbars}
% outer themes include default, infolines, miniframes, shadow, sidebar, smoothbars, smoothtree, split, tree

\useoutertheme[subsection=false]{smoothbars}

% to have the same footer on all slides
%\setbeamertemplate{footline}[text line]{STUFF HERE!}
\setbeamertemplate{footline}[text line]{} % makes the footer EMPTY

% Change some colors:
\setbeamercolor{structure}{fg=CURed}
\setbeamercolor*{palette quaternary}{fg=white,bg=CUGrey}

% include packages
\usepackage{multicol}
\usepackage[all,knot]{xy}
\xyoption{arc}
\usepackage{url}
\usepackage{hyperref}
\usepackage{picture}
%\usepackage{bmpsize}
     
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%% Title Page Info %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\title{Data-driven Whole-Cell Metabolic Modeling}
\subtitle{An Efficient Systems Approach for Flux Estimation}
\author{\texorpdfstring{Brandon Barker\newline
  \url{brandon.barker@gmail.com}}{Brandon Barker}}
\institute{
Tri-Institutional Program in Computational Biology and Medicine\\ 
Gu Laboratory\\
Cornell University \\ \vspace{.25cm}CAC Job Search}
\date{\today}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%% Begin Your Document %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


\begin{document}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
                                       %
\newboolean{thesisStyle}               %
\setboolean{thesisStyle}{true}         %
                                       %
\input{\commonDir documentHeadCommon}  %
                                       %
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%\newcommand\Small{\fontsize{9}{9.2}\selectfont}
%\newcommand*\LSTfont{\scriptsize\ttfamily\SetTracking{encoding=*}{-60}\lsstyle}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\frame{
  \titlepage 
}

\note{
Good morning/thanks/whatever. It has been good meeting everyoe
so far and I look forward to talking with most/all of you later in the
day. I'm a studeint in the Tri-Institutional program in
Computational Biology and Medicine, which has associations
with the Weill Cornell Medical School and Memorial Sloan-Kettering
campuses in NYC.  I'm graduating from Zhenglong Gu's laboratory in 
Nutritional  Sciences - one of the more computational groups in Biology 
at Cornell. Our goal in this research and much of my thesis is
to build tools enabling the computation of a fundamental metabolic,
quantity, flux, in models that may help us to understand disease as
well as various microbial interactiosn, which are somewhat easier
to tackle. In this presentation I will mainly be discussing a recent project
that we've been working on for about a year and a half related to
modeling metabolism of whole cells, but before we get in to that, I
just want to briefly introduce my computational history.
}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% this puts the outline before EACH section automatically & will
% highlight the section you're about to talk about
%\section[Outline]{}
%\frame{\tableofcontents}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


%%%%%%%%%%%%%%%%%%%%%%%%%
                        %
\section{Introduction}  %
\subsection{GPR rules}  %
                        %
%%%%%%%%%%%%%%%%%%%%%%%%%


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\frame{
\frametitle{My computing background}

\begin{itemize}
  \item High school
  \begin{itemize}
    \scriptsize
    \item BSD/Linux sys admin in high school at local ISP.
    \item Learned distributed computing concepts (Bell Labs' Plan9 OS). 
    \item Self-taught first language (x86 assembly).
  \end{itemize}
  \item College activities
  \begin{itemize}
    \scriptsize
    \item Microsoft student ambassador in college:\\
    Promoted, taught, and provided resources for .NET.
    \item IEEE: communications officer, robotics competition (runner-up).
    \item Organized ACM programming competitions.
  \end{itemize}
  \item Bioinformatics IT specialist
  \begin{itemize}
    \scriptsize
    \item Installed OS X cluster with MPI.
    \item Maintained, extended, and taught Microarray database (Oracle SQL/Perl).
    \item OpenMP parallel sequence alignment (C++).
    \item UNIX scripting for comparative genomics.
  \end{itemize}
  \item{Graduate research assistant}
  \begin{itemize}
    \scriptsize
    \item Install \& maintain three servers (Linux/Solaris) for ten active users.
    \item Parallel computing (MATLAB, Python, ATS/C).
    \item Mathematical modeling (convex optimization and dynamical systems).
  \end{itemize}
\end{itemize}

}

\note{\scriptsize
My computing experience began in high school after installing Linux.
During my internship at a local ISP, I gained experience installing
web servers running mostly FreeBSD in a professional setting, and
spent additional time at home learning operating systems concepts and
skills. In college, I participated in various organizatios, including
the ACM, IEEE. I also was a proponent of .NET at the time, and worked
with Microsoft as the student representative on our campus for
teaching .NET and helping students get access to Microsoft software.
\par

I'd always been interested in nature since I grew up in a rural area,
and this later transitioned into interest in biology. I picked up
Biology again more seriously near the end of my undergraduate studies
 and began working as a bioinformatics cluster administrator, which soon
transitioned into a full time position that included maintaining a
database, training its users, and extending its features. The most
prominent feature I added was a K-means clustering web-interface. I
also developed a novel algorithm for biological sequence alignment
that allowed arbritrary precision parameters (using libGMP) and
non-embarassingly parallel execution (using OpenMP). I also did some
comparative genomics research that heavily relied on the UNIX style of
data processing.
\par

Once I transition to graduate research, I couldn't really give up the
sys-admin bug in me, and convinced my advisor to buy one server after
another, which I realize may run counter to the CAC's business
model. But it let me keep my skills sharp. My studies in graduate
school focused on optimization and mathematical modeling, 
which were helpful in designing a highly efficient and more accurate
algorithm which I'll be focusing on today.
\par
}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\frame{\frametitle{Important metabolic pathways}
\begin{center}
\includegraphics[height=0.85\textheight,width=0.53\textwidth]
{metabolism-zoom-start}
\end{center}
}


\note{\footnotesize
One way to describe metabolism is as the set of reactions
and the rate of the reactions made possible by enzymatic catalysts.
Metabolism is critical for understanding wide classes of
diseases including diabetes, cancer, and inborn errors in metabolism.
Here we can see several very important reactions that organisms from
yeasts to humans require, the ultimate step being the generation of
ATP by this motor-like enzyme complex called ATP Synthase. We'll come
back to ATP synthase in a later example.
\par

Unfortunately, metabolism is much more complex than is often
depicted, and so the problem of determining fluxes that might help
us to better understand disease is also difficult problem.
This depiction only shows a little over half of what would
be considered a fairly typical length for a pathway.  This pathway is
inherently linked to another pathway of about the same size, and that
pathway depends on another core pathway of roughly the same length for
preprocessing glucose. When we zoom out, we can start to see some of
this complexity.
\par}


\frame{\frametitle{Important metabolic pathways}
\begin{center}
\includemedia[
  activate=pageopen,
  height=0.85\textheight,
  width=0.53\textwidth,
  transparent=true
]
{}{./figures/metabolism-zoom.swf}
\end{center}
}

\note{Here you can see the rest of the first pathway, oxidative
phosphorylation, and its sister pathway, the TCA cycle. And this
little linear pathway is glycolysis, which converts glucose into a
usable form for the TCA cycle. In practice, these pathways also
require many other pathway dependencies that process or recycle other
chemicals that are not quite in the right form to be used by the later
stages. What's more, this poster from 2003 is actually relatively
simple: chemicals that are similar are often omitted, and there can be
thousands of these for some classes. Also, reactions are present but
are thought to be unimportant often are not drawn. }


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


\frame{\frametitle{An overview to constraint-based modeling}

\newsavebox{\hideLetter}
\savebox{\hideLetter}{%
{\colorbox{white}{\makebox(0.015\textwidth, 0.03\textheight)[lt]{ }}}%
}

\newsavebox{\toynet}
\savebox{\toynet}{%
\includegraphics*[width=0.6\textwidth, viewport=0 418 708 718, clip=true]
{SmatExample}}

\newsavebox{\toySmat}
\savebox{\toySmat}{%
\includegraphics*[width=0.6\textwidth, viewport=0 0 708 418, clip=true]
{SmatExample}}

\newsavebox{\FBAeqns}
\savebox{\FBAeqns}{%
\includegraphics*[width=0.4\textwidth, viewport=411 793 643 938, clip=true]
{SmatExample}}

\newsavebox{\FBAgeom}
\savebox{\FBAgeom}{%
\includegraphics*[width=0.4\textwidth, viewport=0 712 516 1130, clip=true]
{SmatExample}}

\newsavebox{\WIREsCite}
\savebox{\WIREsCite}{\scriptsize%
\textit{Shestov, Barker, Gu, and Locasale. WIREs Syst Biol Med, 2013.}}

% % % %

\begin{picture}(\textwidth, \textheight)
\put(0, \textheight){\makebox(0.5\textwidth, 0)[lt]{\usebox{\toynet}}}
\put(0, 0.95\textheight){\usebox{\hideLetter}}

\put(0, 0.6\textheight){\makebox(0.5\textwidth, 0)[lt]{\usebox{\toySmat}}}
\put(0, 0.555\textheight){\usebox{\hideLetter}}

\put(0.65\textwidth, \textheight){\makebox(0.5\textwidth, 0)[lt]{\usebox{\FBAeqns}}}

\put(0.65\textwidth, 0.6\textheight){\makebox(0.5\textwidth, 0)[lt]{\usebox{\FBAgeom}}}
\put(0.65\textwidth, 0.555\textheight){\usebox{\hideLetter}}
\put(0.98\textwidth, 0.31\textheight){
  \colorbox{white}{\makebox(0.07\textwidth, 0.11\textheight)[lt]{ }}}

\put(0, 0.1\textheight){\usebox{\WIREsCite}}


\end{picture}
} % end of \frame


\note{\scriptsize
To work with a system computationally, we write all known
reactions down, including transport reactions across compartments, as
the columns of a matrix, called the Stoichiometric matrix or S matrix.
The rows correspond to compartmentally unique chemicals, often called
metabolites in the context of metabolism. The entries of the matrix
correspond to the coefficients in the reaction equation: a negative
coefficient denotes it is a reactant, and a positive coefficient
signifies it is a product. For instance, reaction $F_1$ consumes two
$I$ and one $J$ to produce an $A$. We can see that the small toy
network depicted here begins to generate a fair-sized matrix.
\par

Now that we are armed with this matrix, we can make mathematical
assumptions about the system, such as the steady state condition shown
here, which asserts that concentrations do not change over large
enough time scales. In symbols, the S matrix multiplied by the flux
vector yields the change in concenrtation over time for each
metabolite, which is constrained to be zero.  Finally, we need an
optimization objective, such as the maximization of growth rate, which
is common objective used for many microbes.
\par

While this framework is powerful, it is also theoretically simplistic
(so far we only make use of a simple objective and known
reactions). It can also be inaccurate, especially for multicellular
organisms with many different cell types. But the framework is
extendable, and in this work, we extend this framework by using two
types of data: annotation on how the enzymatic complexes come
together, and data on abundance of the components of these enzyme
complexes.
\par}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\frame{\frametitle{An objective for fitting fluxes.}

\begin{block}{Flux fitting}
Started as a way to fit a perturbed model to a wild-type flux, \\
but can also be used to fit empircal flux estimations in a model.\\
Formulated as quadratic least-squares optimization\\(Segr\`e et al 2002):\\
\begin{center}
$\textnormal{minimize}\ \sum\limits_{i=1}^N (v_i-a_i)^2$ 
\end{center}
Later formulated as a linear programing (least absolute devation, or LAD) variant.
\end{block}

}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\frame{\frametitle{From expression to enzyme complex abundance.}

\begin{block}{Expression-flux fitting}
Expression is widely available, especially for diseases. \\
But, expression must be converted to enzyme abundance. 
\end{block}

\begin{figure}
\centering
\includegraphics[clip=true,trim=0cm 0cm 0cm 0cm, width=6cm]{2F43}
\caption{Illustration of the $F_1$ part of the ATP Synthase complex
(PDB ID 1E79).}
\label{fig:2F43}
\end{figure}

}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\frame{\frametitle{Equivalent GPR rules can be evaluated differently.}

\begin{block}{Let's look at an example.}
For genes A, B, and C:\\
Rules $r_1$ and $r_2$ are equivalent logically.\\
However, they can be interpreted differently.
\end{block}

\begin{center}
\includegraphics[width=0.8\textwidth]{GPRrules_diagram}
\vspace{1ex}
\begin{tabular}{cccccccc}
$r_1$ & := & [A and B] or [A and C] & $\rightarrow$ & $e_1$  &=& $\min(a,b$) + $\min(a,c$) \\ 
$r_2$ & := & [A and (B or C)]       & $\rightarrow$ & $e_2$  &=&  $\min(a, b + c$) 
\end{tabular} 
\end{center}

}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section{Methods}         % 
\subsection{Complexation} %
%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{frame}[fragile]
\frametitle{The pipeline and primary interfaces.}

\begin{columns}
\begin{column}{0.45\textwidth}
\begin{figure}
\centering
\begin{tikzpicture}[scale=0.5, every node/.style={scale=0.5}]%[scale=0.8, node distance = 1cm, auto]
    % Place nodes
    \node [block] (start) {start}; 
    \node [iogram, below of=start, left of=start, xshift=-0.2cm] (exp) {Genes:\\ expression 
      ($\mu$,~$\sigma$)}; 
    \node [iogram, below of=start, right of=start] (rules) {Reactions:
      GPR Rule}; 
    \node [block, below of=rules, onslide=<1>{fill=yellow!50}] (parse) {Parse Rule}; 
    \node [block, below of=parse, left of=parse, xshift=-0.5cm, onslide=<2>{fill=yellow!50}]
      (mindisj) {Find minimum disjunction};
    \node [iogram, below of=mindisj] (expstd)
          {Reactions (enzyme~complexes):\\ abundance ($\mu$,~$\sigma$)};
    \node [iogram, right of=falcon, below of=expstd, xshift=0.2cm] (smat) 
      {$\mathbf{S}$ matrix};
    \node [iogram, left of=falcon, below of=expstd, xshift=-0.3cm] (vbnd) 
      {Reactions:\\ flux bounds ($\mathbf{v}_{lb}$, $\mathbf{v}_{ub}$)};
    \node [block, below of=vbnd, left of=smat, right of=vbnd, 
      below of=expstd, yshift=-0.5cm, onslide=<3>{fill=yellow!50}] (falcon) 
      {Flux fitting (FALCON)};
    \node [iogram, below of=falcon] (fluxout) 
      {Reactions:\\ fluxes ($\mathbf{v}:$ $\mu$,~$\sigma$)};
    % Draw edges
    \path [line] (start) -- (exp);
    \path [line] (start) -- (rules);
    \path [line] (rules) -- (parse);
    \path [line] (exp.south) -- (mindisj);
    \path [line] (parse) -- (mindisj);
    \path [line] (mindisj) -- (expstd);
    \path [line, transform canvas={xshift=0.1cm}] (expstd) -- (falcon);
    \path [line] (rules.east) |- (falcon.east);
    \path [line] (smat) -- (falcon);
    \path [line] (vbnd) -- (falcon);
    \path [line] (falcon) -- (fluxout);
\end{tikzpicture}
\end{figure}
\end{column}

\begin{column}{0.55\textwidth}

{\color{gray!80}

{\only<1>{\color{black}}
\begin{lstlisting}[basicstyle=\ttfamily\tiny, language=ATS, showstringspaces=false]


fun parse_fileref (inp: FILEref): grexplst
fun parse_grexplst (!tokener): grexplst
fun parse_grexp (!tokener): grexp


\end{lstlisting}}

{\only<2>{\color{black}}
\begin{lstlisting}[basicstyle=\ttfamily\tiny, language=ATS, showstringspaces=false]



fun grexplst_cnfize (gxs: grexplst): grcnflst
// Easily done with templates:
implement
grexplst_cnfize (gxs) =
  list_map_fun<grexp><grcnf> (gxs, grexp_cnfize)
//
fun grexp_cnfize (gx: grexp): grcnf
//
fun grcnf_minmean_stdev
  (grcnf: !grcnf, emap: GDMap, smap: GDMap): expvar
\end{lstlisting}}

\vspace{4ex}

{\only<3>{\color{black}}
\begin{lstlisting}[basicstyle=\ttfamily\tiny, language=MATLAB, showstringspaces=false]
function [e, e_sd, rxn_rule_group] = ...
    computeMinDisj(model, genedata_filename)
function v = falcon(model, e, e_sd, varargin)
\end{lstlisting}}

} % end of \color{gray!80}
\end{column}

\end{columns}

\end{frame}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\frame{\frametitle{An example of a larger GPR rule in Human Recon 2}

\tiny

%%%%%%%%%%%%%%%%%%%%%%%%%
                        %
\input{exampleRule.tex} %
                        %
%%%%%%%%%%%%%%%%%%%%%%%%%

\note{Now that we have this framework in hand, you might wonder why we
went to the trouble of developing a parser. Some rules can be quite
large. There are several examples in the human model, such as this
one, that won't even fit on a page given a tiny font. In fact, it
would take tens of of pages. the reason for this is due to having
alternative proteins fulfulling identical roles within a large
comoplex.}

}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{frame}[fragile]
\frametitle{GPR API examples}

\begin{columns}
%
\begin{column}{0.5\textwidth}
\lstinputlisting[basicstyle=\ttfamily\tiny, language=ATS, showstringspaces=false]
{falcon_main_fun.dats}
\end{column}
%
\vrule{}
%
\begin{column}{0.5\textwidth}
\lstinputlisting[basicstyle=\ttfamily\tiny, language=ATS, showstringspaces=false]
{falcon_example.dats}
\end{column}
%
\end{columns}
\end{frame}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\frame{\frametitle{Grouping reactions by enzyme complex has a large effect} 

\begin{figure}
\centering
  \includegraphics[height=0.7\textheight]
  {falconGrp_yeastMC}
\caption{
Comparison of setting FALCON to use no reaction group information (x-axis)
versus with group information (y-axis; default FALCON setting).}
\label{fig:FalconGrp}
\end{figure}
}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\frame{\frametitle{Novel features in FALCON flux-fitting}

\begin{itemize}
  \item Automatic expression scaling
    \begin{itemize}
      \footnotesize
      \item Expression scaling 'constant' is now a variable in the L(F)P.
      \item Introduces less bias in a scaling constant.  
    \end{itemize}
  \item Grouping reactions by enzyme complex.
    \begin{itemize}
      \footnotesize
      \item Much more principled approach.
      \item Major difference for some reactions\\
            ($\approx$60-fold difference in some cases).
    \end{itemize}
  \item Batch evaluation of reaction directionality.
    \begin{itemize}
      \footnotesize
      \item Primary reason for increased performance.
      \item Introduces less bias in first stages of directionality assignment.
    \end{itemize}
  \item Take into account nondeterministic nature of algorithms.

\end{itemize}

}

\note{\scriptsize

The scaling constant should be a major help in principle, since it
isn't biased toward any single reaction. Previously, expression was
scaled so that a single reaction's complex abundance and forced flux
would be equal. This is an assumption that doesn't always make sense,
due to the many to one relationship for some reactions and enzyme
complexes. Now, we no longer need to do this is, and relatedly, we no
longer need to force flux through any particular reaction by assuming
it has a non-zero flux.
\par

Reaction grouping is a principled improvement where we directly model
the many-to-one relationship of reactions to enzyme complexes, so that
flux of all the reactions corresponding to one enzyme complex are
summed into one quantity when compared to the enzyme complex's
abundance.
\par

Batch evaulation is another major benefit as it increases the
performance enough to allow many perturbations to be peformed with
ease on microbial models, and it has allowed simulating human cells to
become much more practical (we're down from about an hour to three and
a half minutes). Less bias is introduced in batch evaluation, as 
the order of assignment of directionality can influence later
iterations of the algorithm.
\par

While both our method and a prior method are somewhat
nondeterministic, we added capabilities to both methods to take this
nondeterminism into account, which are represented by the error bars
that you have seen in the scatter plots.
\par
}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\frame{\frametitle{Complexation effect in flux mapping} 

\begin{figure}
\centering
\begin{tabular}{cc}
  \begin{subfigure}[b]{0.5\textwidth}
  \includegraphics[width=\textwidth, trim=9cm 0cm 9cm 0cm, clip=true]
  {expCmpY7MC}
  \caption{yeast} \label{fig:EnzAbundEval:B}
  \end{subfigure} 
&
  \begin{subfigure}[b]{0.5\textwidth}
  \includegraphics[width=\textwidth, trim=9cm 0cm 9cm 0cm, clip=true]
  {expCmpR2dC}
  \caption{human} \label{fig:EnzAbundEval:D}
  \end{subfigure} 
\\
\end{tabular}
\vspace{-4mm}
\label{fig:EnzAbundEval}
\end{figure}
}



%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%%%%%%%%%%%%%%
\section{Results}        %
\subsection{Performance} %
%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\frame{\frametitle{Performance} 

\begin{figure}
\newcommand{\hlY}{\cellcolor{Yellow}}
\centering
\resizebox{\textwidth}{!}{%
\begin{tabular}{l*{7}{r}>{\columncolor{Green}}r}
\textbf{(a)} \hspace{1.2cm} & Max. $\mu$ & Model & 
  Standard FBA & Fitted FBA & GIMME & iMAT & Lee et al.      & FALCON\\
 & 75 \%& Yeast~5 MC & 0.66 & 0.66 & NaN  & 0.57 & \hlY 0.64 & 1 \\
 & 75 \%& Yeast~7 MC & 0.66 & 0.66 & 0.68 & 0.66 & \hlY 0.66 & 0.98\\
 & 75 \%& Yeast~5 HC & 0.73 & 0.78 & 0.75 & 0.66 & 0.98      & 0.99\\
Pearson's r  
 & 75 \%& Yeast~7 HC & 0.70 & 0.70 & 0.80 & 0.66 & 0.98      & 0.99\\
 & 85 \%& Yeast~7 MC & 0.62 & 0.62 & 0.65 & 0.62 & \hlY 0.62 & 0.97\\
 & 85 \%& Yeast~5 HC & 0.88 & 0.89 & 0.9  & 0.81 & 0.99      & 0.99\\
 & 85 \%& Yeast~7 HC & 0.67 & 0.67 & 0.87 & 0.62 & 0.98      & 0.98\\
\end{tabular}}
\caption{Performance of FALCON and other CBM methods for predicting
yeast exometabolic fluxes in two growth conditions (75~\% and 85~\%
Max.~$\mu$)}
\label{tab:FalcPerf}
\end{figure}
}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\frame{\frametitle{Speed} 

\begin{figure}
\centering
\resizebox{\textwidth}{!}{%
\begin{tabular}{l*{6}{r}>{\columncolor{Yellow}}r>{\columncolor{Green}}r}
\textbf{(b)} \hspace{1.2cm} & Max. $\mu$ & Model & 
  Standard FBA & Fitted FBA & GIMME & iMAT & Lee et al. & FALCON\\
 & 75 \%& Yeast~5 MC & 0.9  & 470   & 0.81  & 50     & 110 & 1.8 \\
 & 75 \%& Yeast~7 MC & 1.9  & 3,100 & 2.1   & 12,000 & 600 & 5.6 \\
 & 75 \%& Yeast~5 HC & 0.12 & 110   & 0.18  & 1.4    & 15  & 0.27\\
Time (s) 
 & 75 \%& Yeast~7 HC & 0.72 & 940   & 1.7   & 240    & 670 & 5.5 \\
 & 85 \%& Yeast~7 MC & 2.3  & 3,100 & 3.8   & 14,000 & 610 & 4.6 \\
 & 85 \%& Yeast~5 HC & 0.12 & 110   & 0.18  & 2.5    & 15  & 0.22\\
 & 85 \%& Yeast~7 HC & 0.70 & 110   & 2.5   & 100    & 530 & 5.9\\
\end{tabular}}
\caption{Performance of FALCON and other CBM methods for predicting
yeast exometabolic fluxes in two growth conditions (75~\% and 85~\%
Max.~$\mu$)}
\label{tab:FalcSpeed}
\end{figure}
}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\frame{\frametitle{Permuted expression validates fitting method} 

\begin{figure}
\centering
  \includegraphics[width=\textwidth]{pExp_dirr}
\caption{PDFs of correlation between experimental fluxes and fluxes
estimated from FALCON when all gene expression data points are
permuted.}
\label{fig:YpermCorr}
\end{figure}
}

\note{ 
We want to be certain that the algorithm actually depends on the input
data in a meaningful way, and that we aren't just getting lucky.  To
demonstrate this, on the far right, you can see the FALCON
correlations for two of the conditions in the previous slide. When we
randomly permute the input data thousands of times, we can feed the
permuted expression data into FALCON and see if the correlation
between FALCON-estimated and experimentally-estimated fluxes are
close.  Thankfully, we can see that is not the case, as the
distribution of fluxes is moderately anti-correlated with the
distribution of the rightly estimated flux.
}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\frame{\frametitle{An example of noise in the input data} 
\begin{figure}
\centering
  \includegraphics[width=\textwidth]{proteomicComparison}
\caption{Various scatters for two human proteomic datasets for the
same condition.}
\label{fig:protCmp}
\end{figure}
}

\note{ In almost every type of biological data studied today, there is
a certain degree of error. But in high throughput data, even very
noisy data is valuable because there is so much of it. Here is an
example comparison of two proteomic data sets taken from the same cell
types but using different instruments to measure protein
intensity. The important thing to note in these three visulaizations
is that, while there is a correlation, there is also a fair amount of
noise. We need to make sure our algorithm is capable of handling a
reasonable amount of noise.}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\frame{\frametitle{Sensitivity to expression noise} 
\begin{figure}
\centering
  \includegraphics[width=\textwidth]{noise_y7MCflux}
\caption{Correlation of flux vectors estimated from perturbed versus
unperturbed complex abundance for the Yeast~7 model.}
\label{fig:ExpSens}
\end{figure}
}

\note{In the previous slide, we saw an $r$ of 0.7 for the input
data. This corresponds roughly to a $\sigma$ of 1.25, so that on
average, we might expect an $r$ of 0.6 in the output data, or
flux. But, it is obvious that the worst case is much worse. However,
the good news is that we can take model annotations of known
constraints and reduce the sensitivity even further.
}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\frame{\frametitle{Sensitivity to expression noise: Highly Constrained} 
\begin{figure}
\centering
  \includegraphics[width=\textwidth]{noise_y7HCflux}
\caption{Correlation of flux vectors estimated from perturbed versus
unperturbed complex abundance for the Yeast~7 model.}
\label{fig:ExpSensH}
\end{figure}
}

\note{In this example for yeast, we restored all of the known reaction
directions from the model, and we now see that for a $\sigma$ of 1.25,
we can expect an $r$ of nearly 0.95, and the number of poor
correlations in output at this $\sigma$ level is much less than in the
previous slide. We wanted to test both constraint sets because many
models are not as well understood as yeast, and additionally, reaction
direction may change depending on environmental conditions, though for
many enzymes, it is often safe to assume a given reaction direction.
}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
                            %
\section{Future Directions} %
                            %
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\frame{
\frametitle{Large-scale coupled simulations will be important} 

\begin{columns}
%
\begin{column}{0.35\textwidth}
\begin{itemize}
  \footnotesize
  \item Expression is available for nearly all cell types.
  \item Can simulate tissues, microbial ecosystems, and multicellular organisms.
  \item Amenable to distributed computing.
  \begin{itemize}
    \scriptsize
    \item Cells only need to talk to neighboring cells.
    \item Only a small vector of concentration data needs needs to be passed.
  \end{itemize}
\end{itemize}

\end{column}
%
\begin{column}{0.65\textwidth}
\begin{picture}(200,200)
\centering
\put(0,15){\includegraphics[width=\textwidth]{Figure_S2}}
\put(0,190){\ul{Millions of pairwise simulations reveal}}
\put(0,175){\ul{trends and complex genetic interactions.}}
\put(0,0){\textit{Xu, Barker and Gu. PNAS, 2012.}}
\end{picture}
\end{column}
%
\end{columns}  
}

\note{
*Incomplete*

Bio-energy and industrial bio-synthetics are inherently tied to the
engineering of robust and novel metabolic networks. Metabolism is also
beginning to provide fine-grained detail to understanding microbial
ecologies.
}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\frame{\frametitle{TODO and notes} 

Mention the areas where we parallelized MATLAB code.\\
Slides with many bullet points: grey out not active ones.\\
Maybe use \only http://stackoverflow.com/questions/4683093/beamer-how-to-show-images-as-step-by-step-images\\
Talk about metabolism, but not gene expression, early on. \\
Step through code blocks.\\
Spellcheck!\\
}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\frame{\frametitle{Thank you!} 


\begin{columns}[t]
%
\begin{column}{0.5\textwidth}
\ul{\textit{Collaborators}}
\begin{itemize}
\item Zhenglong Gu
\item Jason W. Locasale
\item Christopher R. Myers
\item Narayanan Sadagopan
\item Kieran Smallbone
\item Yiping Wang
\item Hongwei Xi
\end{itemize}
\end{column}

\begin{column}{0.5\textwidth}
\ul{\textit{Other thanks}}
\begin{itemize}
\item David Christini 
\item Michael Stillman
\item Neil Swainston
\item Lin Xu
\item Tri-I CBM
\item Cornell's CAC
\end{itemize}

\vspace{0.4in}
\TriILogo{0.6in}

\end{column}

%
\end{columns}
}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
                               %
\section[]{Additional Slides}  %
                               %
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\frame{\frametitle{}} %blank slide

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


\frame{\frametitle{Grouping reactions by enzyme complex: HC} 

\begin{figure}
\centering
  \includegraphics[height=0.65\textheight]
  {falconGrp_yeastHC}
\caption{
Comparison of setting FALCON to use no reaction group information (x-axis)
versus with group information (y-axis; default FALCON setting).}
\label{fig:FalconGrpH}
\end{figure}
}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\frame{\frametitle{Permuted expression validates fitting method: HC} 

\begin{figure}
\centering
  \includegraphics[width=\textwidth]{pExp}
\caption{PDFs of correlation between experimental fluxes and fluxes
estimated from FALCON when all gene expression data points are
permuted.}
\label{fig:YpermCorrH}
\end{figure}
}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


\frame{\frametitle{Global flux correlations from permuted inputs} 

\begin{figure}[!htb]
\centering
\begin{tabular}{cc}
  \begin{subfigure}[b]{0.5\textwidth}
  \includegraphics[width=\textwidth]{pFluxVec}
  \caption{highly constrained} \label{fig:YpermCorrSup:A}
  \end{subfigure}
&
  \begin{subfigure}[b]{0.5\textwidth}
  \includegraphics[width=\textwidth]{pFluxVec_dirr}
  \caption{minimally constrained} \label{fig:YpermCorrSup:B}
  \end{subfigure} 
\\
\end{tabular}
\vspace{-4mm}
\caption{PDFs of Pearson's correlation between fluxes estimated from
permuted and unpermuted complex abundance.}
\label{fig:YpermCorrSup:pearson}
\end{figure}
}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\frame{\frametitle{Global flux correlations from permuted inputs} 

\begin{figure}[!htb]
\centering
\begin{tabular}{cc}
  \begin{subfigure}[b]{0.5\textwidth}
  \includegraphics[width=\textwidth]{kFluxVec}
  \caption{highly constrained} \label{fig:YpermCorrSup:C}
  \end{subfigure}
&
  \begin{subfigure}[b]{0.5\textwidth}
  \includegraphics[width=\textwidth]{kFluxVec_dirr}
  \caption{minimally constrained} \label{fig:YpermCorrSup:D}
  \end{subfigure} 
\\
\end{tabular}
\vspace{-4mm}
\caption{PDFs of Kendall's correlation between fluxes estimated from
permuted and unpermuted complex abundance.}
\label{fig:YpermCorrSup:kendall}
\end{figure}
}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\frame{\frametitle{Correlation of perturbed inputs (complex abundance)} 

\begin{figure}
\centering
\includegraphics[width=\textwidth]{noise_y7EC}
\caption{Correlation of perturbed enzyme abundance vectors with the
associated unperturbed vector for the Yeast~7 model.}
\label{fig:ExpSens:Complex}
\end{figure}

}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


\frame{\frametitle{Flux estimates give us more than just complex abundance} 

\hl{advantage of looking at flux over expression, which is commonly}
\hl{move this figure out and cite correlation}

\begin{figure}
\begin{tabular}{cc}
  \begin{subfigure}[b]{0.5\textwidth}   % l   b   r   t
  \includegraphics[width=\textwidth, trim=9cm 1.2cm 9cm 1cm, clip=true]
    {YeastExpFluxCompare}
  \caption{yeast} \label{fig:FluxExpCmp:A}
  \end{subfigure}
&
  \begin{subfigure}[b]{0.5\textwidth}
  \includegraphics[width=\textwidth, trim=9cm 1.2cm 9cm 1cm, clip=true]
    {HumanExpFluxCompare}
  \caption{human} \label{fig:FluxExpCmp:B}
  \end{subfigure}
\\
\end{tabular}
\caption{Pearson correlation between FALCON flux magnitudes, and
various complexation estimates in yeast \textbf{(a)} and human
\textbf{(b)}.}
\label{fig:FluxExpCmp}
\end{figure}
}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%% End Your Document %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\end{document}

