\newcommand{\captionpage}[2]{
\newpage
\vspace*{\fill}
\begin{center}
\captionof{#1}{#2}
\end{center}
\vspace{\fill}
\newpage
}


% Define block styls
\tikzstyle{line} = [draw, very thick, color=black!50, -latex']
\tikzstyle{cloud} = [draw, ellipse,fill=red!20, node distance=2em]
\tikzstyle{decision} = [diamond, draw, fill=blue!20, align=center,
    text badly centered, node distance=6em, inner sep=0pt]
\tikzstyle{block} = [rectangle, draw, %fill=blue!20,
    text centered, align=center, node distance=4em]
\tikzstyle{iogram} = [trapezium, draw, fill=black!10, %fill=pink!20,
    trapezium left angle=70, trapezium right angle=-70,
    text centered, align=center, node distance=5em]
\tikzset{
  onslide/.code args={<#1>#2}{\only<#1>{\pgfkeysalso{#2}}}, 
}

% Some global commands for alternatives

\DeclareRobustCommand\suppOrApp{%
  \ifthenelse{\boolean{thesisStyle}}%
    {}%
    {Supplementary}%
}
\DeclareRobustCommand\Fig{%
  \ifthenelse{\boolean{thesisStyle}}%
    {Figure}%
    {Fig.}%
}



% \ifthenelse{\boolean{thesisStyle}}
% { 
% % \renewcommand{\caption}[1]{\singlespacing\hangcaption{#1}\normalspacing}
%   \renewcommand{\topfraction}{0.85}
%   \renewcommand{\textfraction}{0.1}
% %
% % This seems to cause a problem with floats and excess vertical space:
% %
%   \renewcommand{\floatpagefraction}{0.75}
% }
% {}

\ifthenelse{\boolean{thesisStyle}}{%
  \floatstyle{plaintop}
  \restylefloat{table}
}
{}


% \epstopdfDeclareGraphicsRule{.eps}{pdf}{.pdf}{%
%   epstool --copy --bbox #1 --output epstool_tmp.eps;
%   epstopdf --hires epstool_tmp.eps --outfile \OutputFile;
%   rm epstool_tmp.eps
% }


\def\D{\mathrm{d}}

%\newcommand{\ANDw}{\textnormal{AND}}
%\newcommand{\ORw}{\textnormal{OR}}
\newcommand{\ANDw}{\land}
\newcommand{\ORw}{\lor}


%%%%%%%%%%%%%%%%%%%%%%%%%%%% Introduction %%%%%%%%%%%%%%%%%%%%%%%%%%%%
\newcommand{\introSameGeneCredit}{%
\ifthenelse{\boolean{thesisStyle}}{%
  \footnote{This chapter is taken from material in \citealt{Shestov2013a}.
Brandon Barker is the primary author of all material found herein.}%
}%
{}%
}


%% Dynamic Epistasis for Different Alleles of the Same Gene %%

\newcommand{\epiSameGeneCredit}{%
\ifthenelse{\boolean{thesisStyle}}{%
  \footnote{This chapter is published as \citet{Xu2012}.
Brandon Barker and Lin Xu contributed equally to this work.
It is additionally available in \citet[chapter 4]{Xu2012a}.}%
}%
{}%
}

\newcommand{\epiSameGeneAbstract}{%
Epistasis refers to the phenomenon in which phenotypic consequences
caused by mutation of one gene depend on one or more mutations at
another gene. Epistasis is critical for understanding many genetic and
evolutionary processes, including pathway organization, evolution of
sexual reproduction, mutational load, ploidy, genomic complexity,
speciation, and the origin of life. Nevertheless, current
understandings for the genome-wide distribution of epistasis are
mostly inferred from interactions among one mutant type per gene,
whereas how epistatic interaction partners change dynamically for
different mutant alleles of the same gene is largely unknown. Here we
address this issue by combining predictions from flux balance analysis
and data from a recently published high-throughput experiment. Our
results show that different alleles can epistatically interact with
very different gene sets. Furthermore, between two random mutant
alleles of the same gene, the chance for the allele with more severe
mutational consequence to develop a higher percentage of negative
epistasis than the other allele is 50-70\% in eukaryotic organisms,
but only 20-30\% in bacteria and archaea. We developed a population
genetics model that predicts that the observed distribution for the
sign of epistasis can speed up the process of purging deleterious
mutations in eukaryotic organisms. Our results indicate that epistasis
among genes can be dynamically rewired at the genome level, and call
on future efforts to revisit theories that can integrate epistatic
dynamics among genes in biological systems\epiSameGeneCredit.
}

%%%%%%%%%%% Environmental Epistasis with FBA %%%%%%%%%%%

\newcommand{\epistasisEnviroAbstract}{%
Epistasis describes the phenomenon that mutations at different loci do
not have independent effects with regard to certain
phenotypes. Understanding the global epistatic landscape is vital for
many genetic and evolutionary theories. Current knowledge for
epistatic dynamics under multiple conditions is limited by the
technological difficulties in experimentally screening epistatic
relations among genes. We explored this issue by applying flux balance
analysis to simulate epistatic landscapes under various
environmental perturbations. Specifically, we looked at gene-gene
epistatic interactions, where the mutations were assumed to occur in
different genes. We predicted that epistasis tends to become more
positive from glucose-abundant to nutrient-limiting conditions,
indicating that selection might be less effective in removing
deleterious mutations in the latter. We also observed a stable core of
epistatic interactions in all tested conditions, as well as many
epistatic interactions unique to each condition. Interestingly, genes
in the stable epistatic interaction network are directly linked to
most other genes whereas genes with condition-specific epistasis form
a scale-free network. Furthermore, genes with stable epistasis tend to
have similar evolutionary rates, whereas this co-evolving relationship
does not hold for genes with condition-specific epistasis. Our
findings provide a novel genome-wide picture about epistatic dynamics
under environmental perturbations.
}

\newcommand{\epistasisEnviroAuthorSummary}{%
Epistasis, often referred to as genetic interactions, occur when
mutational effects of genes depend on each other. Aside from often
times complicating the way in which the phenotype of an organism
relates to its genotype, epistatic interactions (or epistases) are
essential to several important theories in biology, especially in
evolution. Due to the difficulty in experimentally assessing epistasis
across an entire genome, we employed mathematical modeling of the
metabolic network of bakerâ€™s yeast to comprehensively simulate genetic
interactions for virtually all known metabolic genes in the
organism. We performed comprehensive simulations in 17 different
environments, which differ by their nutrients. We characterized a
trend that occurs in genetic interactions when yeast is transferred
from a glucose-abundant environment to other environments. We also
found that both the set of genetic interactions present in all
conditions and the set of interactions present in a single environment
are fairly large sets with highly different connectivity. Furthermore,
the set present in all conditions tends to consist of gene pairs with
similar evolutionary rates.
}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%% FALCON %%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\newcommand{\falconAbstractMotivation}{%
A major theme in constraint-based modeling is unifying 
experimental data, such as biochemical information about the reactions
that can occur in a system or the composition and localization of enzyme
complexes, with high-throughput data including expression data,
metabolomics, or DNA sequencing. The desired result is to increase
 predictive capability and improve our understanding of metabolism.
 The approach typically employed when only gene (or protein) intensities
are available is the creation of tissue-specific models, which reduces
the available reactions in an organism model, and does not provide an
objective function for the estimation of fluxes.
}

\newcommand{\falconAbstractResults}{%
We develop a method, flux assignment with LAD (least absolute
deviation) convex objectives and normalization (FALCON),
 that employs metabolic network reconstructions along with expression
data to estimate fluxes. In order to use such a method, accurate
measures of enzyme complex abundance are needed, so we first
present an algorithm that addresses quantification of complex
abundance. Our extensions to prior techniques include the
capability to work with large models and significantly improved
run-time performance even for smaller models, an improved analysis of
enzyme complex formation, the ability to handle large enzyme
complex rules that may incorporate multiple isoforms, and either
maintained or significantly improved correlation with experimentally
measured fluxes.
}

\newcommand{\falconAbstractAvail}{%
FALCON has been implemented in MATLAB and ATS, and can be downloaded
from: \url{https://github.com/bbarker/FALCON}. ATS is not required to
compile the software, as intermediate C source code is available. 
FALCON requires use of the COBRA Toolbox, also implemented in MATLAB.
}


%% Beneficial Mutant Epistasis %%

\newcommand{\epiBeneMutAbstract}{%
Existing literature has only dealt with the simulation of strictly
deleterious mutants rather than beneficial mutants in the
constraint-based modeling literature, which is chiefly due to existing
studies optimizing the fitness function, which leaves no room for
improvement. In this study, we develop a constraint-based approach
that can simulate beneficial, neutral, and deleterious mutations.  We
show that this simulation technique can be useful for understanding
adaptive trajectories, and we develop a software library for the
analysis of evolutionary paths.  Our mechanistic model can reproduce
the distribution of
epistases between beneficial mutations that was observed in a data set
and a population genetic model fit to the same data set, showing that
our model behaves appropriately in this conext and may be a useful
tool for further evolutionary analyses. Finally, in experimental data
sets and in our simulations, slightly beneficial mutations are much
more likely to have positive (synergistic) epistasis with other
beneficial mutations, making their likelihood of becoming fixed higher
than would be expected without considering epistatic effects.
}

\captionsetup{labelfont=bf}


% Take care of potentially defined variables:

% \newcommand[1]{\identifndef}{
%   \if\isdef\csname{#1}
%     {}
%   \else
%     \newcommand{\{#1}}[1]{\{#1}}
%   \fi
% }

% \identifndef{processtable}


